// @builder.io/sdk-vue/lib/browser/index-yersTeTV.js
var mt = Object.defineProperty;
var ht = (e, t, n) => t in e ? mt(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var re = (e, t, n) => (ht(e, typeof t != "symbol" ? t + "" : t, n), n);
import "./style.css";
import { defineComponent as f, openBlock as d, createElementBlock as p, Fragment as I, createBlock as k, resolveDynamicComponent as K, mergeProps as S, toHandlers as P, withCtx as W, renderSlot as E, resolveComponent as g, createTextVNode as ft, toDisplayString as ne, defineAsyncComponent as H, createVNode as $, renderList as N, createCommentVNode as v, h as Ae, createElementVNode as z, normalizeStyle as O, normalizeClass as J, markRaw as bt } from "vue";
const gt = /* @__PURE__ */ new Set(["area", "base", "br", "col", "embed", "hr", "img", "input", "keygen", "link", "meta", "param", "source", "track", "wbr"]), yt = (e) => typeof e == "string" && gt.has(e.toLowerCase());
function x(e = {}, t, n) {
  const o = {};
  for (const r in e) {
    if (!e[r] || n && !r.startsWith(t))
      continue;
    const i = n ? r.replace(t, "") : r;
    o[i] = e[r];
  }
  return o;
}
const kt = f({
  name: "dynamic-renderer",
  props: ["TagName", "attributes", "actionAttributes"],
  data() {
    return { isEmptyElement: yt };
  }
}), b = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [o, r] of t)
    n[o] = r;
  return n;
};
function vt(e, t, n, o, r, i) {
  return e.isEmptyElement(e.TagName) ? (d(), k(K(e.TagName), S({ key: 1 }, e.attributes, P(e.actionAttributes)), null, 16)) : (d(), p(I, { key: 0 }, [
    typeof e.TagName == "string" ? (d(), k(K(e.TagName), S({ key: 0 }, e.attributes, P(e.actionAttributes)), {
      default: W(() => [
        E(e.$slots, "default")
      ]),
      _: 3
    }, 16)) : (d(), k(K(e.TagName), S({ key: 1 }, e.attributes, P(e.actionAttributes)), {
      default: W(() => [
        E(e.$slots, "default")
      ]),
      _: 3
    }, 16))
  ], 64));
}
const he = /* @__PURE__ */ b(kt, [["render", vt]]), Ei = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: he
}, Symbol.toStringTag, { value: "Module" })), A = "vue", ee = () => {
  switch (A) {
    case "react":
    case "reactNative":
    case "rsc":
      return "className";
    case "svelte":
    case "vue":
    case "solid":
    case "qwik":
    case "angular":
      return "class";
  }
}, Ct = f({
  name: "builder-button",
  components: { DynamicRenderer: he },
  props: [
    "attributes",
    "link",
    "openLinkInNewTab",
    "builderLinkComponent",
    "text"
  ],
  data() {
    return { filterAttrs: x };
  },
  methods: {
    attrs() {
      return {
        ...x(this.attributes, "v-on:", !1),
        [ee()]: `${this.link ? "" : "builder-button"} ${this.attributes[ee()] || ""}`,
        ...this.link ? {
          href: this.link,
          target: this.openLinkInNewTab ? "_blank" : void 0,
          role: "link"
        } : {
          role: "button"
        }
      };
    }
  }
});
function St(e, t, n, o, r, i) {
  const s = g("DynamicRenderer");
  return d(), k(s, {
    attributes: e.attrs(),
    TagName: e.link ? e.builderLinkComponent || "a" : "button",
    actionAttributes: e.filterAttrs(e.attributes, "v-on:", !0)
  }, {
    default: W(() => [
      ft(ne(e.text), 1)
    ]),
    _: 1
  }, 8, ["attributes", "TagName", "actionAttributes"]);
}
const wt = /* @__PURE__ */ b(Ct, [["render", St]]), Tt = Symbol(), de = {
  Builder: {
    content: null,
    context: {},
    localState: void 0,
    rootSetState() {
    },
    rootState: {},
    apiKey: null,
    apiVersion: void 0,
    componentInfos: {},
    inheritedStyles: {},
    BlocksWrapper: "div",
    BlocksWrapperProps: {}
  },
  key: Tt
}, It = Symbol(), Ve = {
  Components: { registeredComponents: {} },
  key: It
};
function $t(e) {
  var t;
  return {
    ...(t = e.component) == null ? void 0 : t.options,
    ...e.options
  };
}
const Z = "[Builder.io]: ", B = {
  log: (...e) => console.log(Z, ...e),
  error: (...e) => console.error(Z, ...e),
  warn: (...e) => console.warn(Z, ...e),
  debug: (...e) => console.debug(Z, ...e)
};
function T() {
  return typeof window < "u" && typeof document < "u";
}
const Rt = (e) => {
  const t = {};
  return e.forEach((n, o) => {
    t[o] = n;
  }), t;
}, We = (e) => e instanceof URLSearchParams ? Rt(e) : e, Ne = (e) => typeof e == "string" ? e : e instanceof URLSearchParams ? e.toString() : new URLSearchParams(e).toString();
function Bt() {
  return T() && window.self !== window.top;
}
function R(e) {
  return Bt() && // accessing window.location.search is safe here because `isIframe()` is only `true` if we're in a browser.
  Ne(e || window.location.search).indexOf("builder.frameEditing=") !== -1;
}
const Pt = () => {
  if (T()) {
    const e = new URL(location.href);
    return e.pathname === "" && (e.pathname = "/"), e;
  } else
    return console.warn("Cannot get location for tracking in non-browser environment"), null;
}, Et = () => typeof navigator == "object" && navigator.userAgent || "", Fe = () => {
  const e = Et(), t = {
    Android() {
      return e.match(/Android/i);
    },
    BlackBerry() {
      return e.match(/BlackBerry/i);
    },
    iOS() {
      return e.match(/iPhone|iPod/i);
    },
    Opera() {
      return e.match(/Opera Mini/i);
    },
    Windows() {
      return e.match(/IEMobile/i) || e.match(/WPDesktop/i);
    },
    any() {
      return t.Android() || t.BlackBerry() || t.iOS() || t.Opera() || t.Windows() || A === "reactNative";
    }
  }, n = e.match(/Tablet|iPad/i), o = Pt();
  return {
    urlPath: o == null ? void 0 : o.pathname,
    host: (o == null ? void 0 : o.host) || (o == null ? void 0 : o.hostname),
    device: n ? "tablet" : t.any() ? "mobile" : "desktop"
  };
}, xt = ({
  builder: e,
  context: t,
  event: n,
  state: o
}) => Object.entries({
  state: o,
  Builder: e,
  // legacy
  builder: e,
  context: t,
  event: n
}), At = () => ({
  isEditing: R(),
  isBrowser: T(),
  isServer: !T(),
  getUserAttributes: () => Fe()
}), Vt = (e, {
  isExpression: t = !0
}) => /* we disable this for cases where we definitely don't want a return */ t && !(e.includes(";") || e.includes(" return ") || e.trim().startsWith("return ")) ? `return (${e});` : e, Se = ({
  code: e,
  builder: t,
  context: n,
  event: o,
  localState: r,
  rootSetState: i,
  rootState: s
}) => {
  const l = xt({
    builder: t,
    context: n,
    event: o,
    state: Le({
      rootState: s,
      localState: r,
      rootSetState: i
    })
  });
  return new Function(...l.map(([a]) => a), e)(...l.map(([, a]) => a));
};
function Le({
  rootState: e,
  localState: t,
  rootSetState: n
}) {
  return new Proxy(e, {
    get: (o, r) => {
      if (t && r in t)
        return t[r];
      const i = o[r];
      return typeof i == "object" && i !== null ? Le({
        rootState: i,
        localState: void 0,
        rootSetState: n ? (s) => {
          o[r] = s, n(o);
        } : void 0
      }) : i;
    },
    set: (o, r, i) => {
      if (t && r in t)
        throw new Error("Writing to local state is not allowed as it is read-only.");
      return o[r] = i, n == null || n(o), !0;
    }
  });
}
const D = (e) => e != null;
function Wt() {
  var e;
  return typeof process < "u" && D((e = process == null ? void 0 : process.versions) == null ? void 0 : e.node);
}
const Nt = () => {
  var o;
  if (!Wt())
    return !1;
  const e = process.arch === "arm64", t = process.version.startsWith("v20"), n = (o = process.env.NODE_OPTIONS) == null ? void 0 : o.includes("--no-node-snapshot");
  return e && t && !n ? (B.log("Skipping usage of `isolated-vm` to avoid crashes in Node v20 on an arm64 machine.\n    If you would like to use the `isolated-vm` package on this machine, please provide the `NODE_OPTIONS=--no-node-snapshot` config to your Node process.\n    See https://github.com/BuilderIO/builder/blob/main/packages/sdks/README.md#node-v20--m1-macs-apple-silicon-support for more information.\n    "), !0) : !1;
}, Ft = (e) => (T() || Nt(), Se(e)), V = class V {
  static getCacheKey(t) {
    return JSON.stringify({
      ...t,
      // replace the event with a random number to break cache
      // thats because we can't serialize the event object due to circular refs in DOM node refs.
      event: t.event ? Math.random() : void 0
    });
  }
  static getCachedValue(t) {
    return V.cache.get(t);
  }
  static setCachedValue(t, n) {
    V.cache.size > 20 && V.cache.delete(V.cache.keys().next().value), V.cache.set(t, {
      value: n
    });
  }
};
re(V, "cacheLimit", 20), re(V, "cache", /* @__PURE__ */ new Map());
let U = V;
function te({
  code: e,
  context: t,
  localState: n,
  rootState: o,
  rootSetState: r,
  event: i,
  isExpression: s = !0,
  enableCache: l
}) {
  if (e === "")
    return;
  const a = {
    code: Vt(e, {
      isExpression: s
    }),
    builder: At(),
    context: t,
    event: i,
    rootSetState: r,
    rootState: o,
    localState: n
  };
  if (l) {
    const u = U.getCacheKey(a), h = U.getCachedValue(u);
    if (h)
      return h.value;
  }
  try {
    const u = Ft(a);
    if (l) {
      const h = U.getCacheKey(a);
      U.setCachedValue(h, u);
    }
    return u;
  } catch (u) {
    B.error("Failed code evaluation: " + u.message, {
      code: e
    });
    return;
  }
}
const Y = (e) => JSON.parse(JSON.stringify(e)), Oe = (e, t, n) => {
  if (Object(e) !== e)
    return e;
  const o = Array.isArray(t) ? t : t.toString().match(/[^.[\]]+/g);
  return o.slice(0, -1).reduce((r, i, s) => Object(r[i]) === r[i] ? r[i] : r[i] = Math.abs(Number(o[s + 1])) >> 0 === +o[s + 1] ? [] : {}, e)[o[o.length - 1]] = n, e;
};
const Lt = ({
  block: e,
  context: t,
  localState: n,
  rootState: o,
  rootSetState: r
}) => {
  if (!e.bindings)
    return e;
  const i = Y(e), s = {
    ...i,
    properties: {
      ...i.properties
    },
    actions: {
      ...i.actions
    }
  };
  for (const l in e.bindings) {
    const a = e.bindings[l], u = te({
      code: a,
      localState: n,
      rootState: o,
      rootSetState: r,
      context: t,
      enableCache: !0
    });
    Oe(s, l, u);
  }
  return s;
};
function De({
  block: e,
  context: t,
  shouldEvaluateBindings: n,
  localState: o,
  rootState: r,
  rootSetState: i
}) {
  const s = e;
  return n ? Lt({
    block: s,
    localState: o,
    rootState: r,
    rootSetState: i,
    context: t
  }) : s;
}
const fe = (e) => e ? e.replace(/([a-z0-9]|(?=[A-Z]))([A-Z])/g, "$1-$2").toLowerCase() : "";
function Ot(e, t, n = {}) {
  let o, r, i, s = null, l = 0;
  const a = function() {
    l = n.leading === !1 ? 0 : Date.now(), s = null, i = e.apply(o, r), s || (o = r = null);
  };
  return function() {
    const u = Date.now();
    !l && n.leading === !1 && (l = u);
    const h = t - (u - l);
    return o = this, r = arguments, h <= 0 || h > t ? (s && (clearTimeout(s), s = null), l = u, i = e.apply(o, r), s || (o = r = null)) : !s && n.trailing !== !1 && (s = setTimeout(a, h)), i;
  };
}
function q(e, ...t) {
  const n = Object(e);
  for (let o = 1; o < arguments.length; o++) {
    const r = arguments[o];
    if (r != null)
      for (const i in r)
        Object.prototype.hasOwnProperty.call(r, i) && (n[i] = r[i]);
  }
  return n;
}
function Dt(e) {
  for (const t of e)
    switch (t.trigger) {
      case "pageLoad":
        je(t);
        break;
      case "scrollInView":
        Mt(t);
        break;
    }
}
function Ue(e) {
  console.warn(`Cannot animate element: element with ID ${e} not found!`);
}
function Me(e, t) {
  const n = Ut(e), o = getComputedStyle(t), r = e.steps[0].styles, i = e.steps[e.steps.length - 1].styles, s = [r, i];
  for (const l of s)
    for (const a of n)
      a in l || (l[a] = o[a]);
}
function Ut(e) {
  const t = [];
  for (const n of e.steps)
    for (const o in n.styles)
      t.indexOf(o) === -1 && t.push(o);
  return t;
}
function je(e) {
  const t = Array.prototype.slice.call(document.getElementsByClassName(e.elementId || e.id || ""));
  if (!t.length) {
    Ue(e.elementId || e.id || "");
    return;
  }
  Array.from(t).forEach((n) => {
    Me(e, n), n.style.transition = "none", n.style.transitionDelay = "0", q(n.style, e.steps[0].styles), setTimeout(() => {
      n.style.transition = `all ${e.duration}s ${fe(e.easing)}`, e.delay && (n.style.transitionDelay = e.delay + "s"), q(n.style, e.steps[1].styles), setTimeout(() => {
        n.style.transition = "", n.style.transitionDelay = "";
      }, (e.delay || 0) * 1e3 + e.duration * 1e3 + 100);
    });
  });
}
function Mt(e) {
  const t = Array.prototype.slice.call(document.getElementsByClassName(e.elementId || e.id || ""));
  if (!t.length) {
    Ue(e.elementId || e.id || "");
    return;
  }
  Array.from(t).forEach((n) => {
    Me(e, n);
    let o = !1, r = !1;
    function i() {
      !o && l(n) ? (o = !0, r = !0, setTimeout(() => {
        q(n.style, e.steps[1].styles), e.repeat || document.removeEventListener("scroll", s), setTimeout(() => {
          r = !1, e.repeat || (n.style.transition = "", n.style.transitionDelay = "");
        }, (e.duration + (e.delay || 0)) * 1e3 + 100);
      })) : e.repeat && o && !r && !l(n) && (o = !1, q(n.style, e.steps[0].styles));
    }
    const s = Ot(i, 200, {
      leading: !1
    });
    function l(h) {
      const c = h.getBoundingClientRect(), m = window.innerHeight, y = (e.thresholdPercent || 0) / 100 * m;
      return c.bottom > y && c.top < m - y;
    }
    const a = e.steps[0].styles;
    function u() {
      q(n.style, a);
    }
    u(), setTimeout(() => {
      n.style.transition = `all ${e.duration}s ${fe(e.easing)}`, e.delay && (n.style.transitionDelay = e.delay + "s");
    }), document.addEventListener("scroll", s, {
      capture: !0,
      passive: !0
    }), i();
  });
}
const He = (e) => Object.entries(e).map(([n, o]) => {
  if (typeof o == "string")
    return `${fe(n)}: ${o};`;
}).filter(D), jt = (e) => He(e).join(`
`), xi = ({
  mediaQuery: e,
  className: t,
  styles: n
}) => {
  const o = `.${t} {
    ${jt(n)}
  }`;
  return e ? `${e} {
      ${o}
    }` : o;
};
function Ht({
  style: e
}) {
  return e;
}
const Ai = ({
  block: e,
  context: t
}) => Ke(Ht({
  style: e.style || {},
  context: t,
  block: e
}));
function Ke(e) {
  switch (A) {
    case "svelte":
    case "vue":
    case "solid":
    case "angular":
      return He(e).join(" ");
    case "qwik":
    case "reactNative":
    case "react":
    case "rsc":
      return e;
  }
}
const Kt = ({
  block: e,
  context: t,
  registeredComponents: n
}) => {
  var i;
  const o = (i = De({
    block: e,
    localState: t.localState,
    rootState: t.rootState,
    rootSetState: t.rootSetState,
    context: t.context,
    shouldEvaluateBindings: !1
  }).component) == null ? void 0 : i.name;
  if (!o)
    return null;
  const r = n[o];
  if (r)
    return r;
  console.warn(`
      Could not find a registered component named "${o}". 
      If you registered it, is the file that registered it imported by the file that needs to render it?`);
}, qt = ({
  block: e,
  context: t
}) => {
  const {
    repeat: n,
    ...o
  } = e;
  if (!(n != null && n.collection))
    return;
  const r = te({
    code: n.collection,
    localState: t.localState,
    rootState: t.rootState,
    rootSetState: t.rootSetState,
    context: t.context,
    enableCache: !0
  });
  if (!Array.isArray(r))
    return;
  const i = n.collection.split(".").pop(), s = n.itemName || (i ? i + "Item" : "item");
  return r.map((a, u) => ({
    context: {
      ...t,
      localState: {
        ...t.localState,
        $index: u,
        $item: a,
        [s]: a,
        [`$${s}Index`]: u
      }
    },
    block: o
  }));
}, oe = (e) => ({
  // once we bump to a major version, toggle this to `false`.
  builderBlock: !0,
  // once we bump to a major version, toggle this to `false`.
  builderContext: !0,
  builderComponents: !1,
  builderLinkComponent: !1,
  ...e
}), zt = (e, t) => e ? oe(e.shouldReceiveBuilderProps).builderLinkComponent ? {
  builderLinkComponent: t
} : {} : {}, Jt = (e, t) => e ? oe(e.shouldReceiveBuilderProps).builderComponents ? {
  builderComponents: t
} : {} : {}, Gt = (e, t) => e ? oe(e.shouldReceiveBuilderProps).builderBlock ? {
  builderBlock: t
} : {} : {}, Yt = (e, t) => e ? oe(e.shouldReceiveBuilderProps).builderContext ? {
  builderContext: t
} : {} : {}, Qt = () => Promise.resolve().then(() => un).then((e) => e.default).catch((e) => {
  throw console.error(
    "Error while attempting to dynamically import component DynamicDiv at ../dynamic-div.vue",
    e
  ), e;
}), Xt = () => import("./block-styles-yF029qLO.js").then((e) => e.default).catch((e) => {
  throw console.error(
    "Error while attempting to dynamically import component BlockStyles at ./components/block-styles.vue",
    e
  ), e;
}), Zt = () => import("./block-wrapper-slzYlPm8.js").then((e) => e.default).catch((e) => {
  throw console.error(
    "Error while attempting to dynamically import component BlockWrapper at ./components/block-wrapper.vue",
    e
  ), e;
}), _t = () => import("./component-ref-f280Pij9.js").then((e) => e.default).catch((e) => {
  throw console.error(
    "Error while attempting to dynamically import component ComponentRef at ./components/component-ref/component-ref.vue",
    e
  ), e;
}), en = () => import("./repeated-block-SxyAWZbs.js").then((e) => e.default).catch((e) => {
  throw console.error(
    "Error while attempting to dynamically import component RepeatedBlock at ./components/repeated-block.vue",
    e
  ), e;
}), tn = f({
  name: "block",
  components: {
    BlockStyles: H(Xt),
    ComponentRef: H(_t),
    RepeatedBlock: H(en),
    BlockWrapper: H(Zt),
    DynamicDiv: H(Qt)
  },
  props: ["block", "context", "registeredComponents", "linkComponent"],
  mounted() {
    const e = this.processedBlock.id, t = this.processedBlock.animations;
    t && e && Dt(
      t.map((n) => ({
        ...n,
        elementId: e
      }))
    );
  },
  computed: {
    blockComponent() {
      return Kt({
        block: this.block,
        context: this.context,
        registeredComponents: this.registeredComponents
      });
    },
    repeatItem() {
      return qt({
        block: this.block,
        context: this.context
      });
    },
    processedBlock() {
      var e;
      return (e = this.block.repeat) != null && e.collection ? this.block : De({
        block: this.block,
        localState: this.context.localState,
        rootState: this.context.rootState,
        rootSetState: this.context.rootSetState,
        context: this.context.context,
        shouldEvaluateBindings: !0
      });
    },
    Tag() {
      var t;
      return this.block.tagName === "a" || ((t = this.processedBlock.properties) == null ? void 0 : t.href) || this.processedBlock.href ? this.linkComponent || "a" : this.block.tagName || "div";
    },
    canShowBlock() {
      var n, o;
      if ((n = this.block.repeat) != null && n.collection)
        return !!((o = this.repeatItem) != null && o.length);
      const e = "hide" in this.processedBlock ? this.processedBlock.hide : !1;
      return ("show" in this.processedBlock ? this.processedBlock.show : !0) && !e;
    },
    childrenWithoutParentComponent() {
      var t;
      return !((t = this.blockComponent) != null && t.component) && !this.repeatItem ? this.processedBlock.children ?? [] : [];
    },
    componentRefProps() {
      var e, t, n;
      return {
        blockChildren: this.processedBlock.children ?? [],
        componentRef: (e = this.blockComponent) == null ? void 0 : e.component,
        componentOptions: {
          ...$t(this.processedBlock),
          ...Gt(this.blockComponent, this.processedBlock),
          ...Yt(this.blockComponent, this.context),
          ...zt(this.blockComponent, this.linkComponent),
          ...Jt(
            this.blockComponent,
            this.registeredComponents
          )
        },
        context: this.context,
        linkComponent: this.linkComponent,
        registeredComponents: this.registeredComponents,
        builderBlock: this.processedBlock,
        includeBlockProps: ((t = this.blockComponent) == null ? void 0 : t.noWrap) === !0,
        isInteractive: !((n = this.blockComponent) != null && n.isRSC)
      };
    }
  }
});
function nn(e, t, n, o, r, i) {
  var c;
  const s = g("BlockStyles"), l = g("ComponentRef"), a = g("Block", !0), u = g("BlockWrapper"), h = g("RepeatedBlock");
  return e.canShowBlock ? (d(), p(I, { key: 0 }, [
    $(s, {
      block: e.block,
      context: e.context
    }, null, 8, ["block", "context"]),
    (c = e.blockComponent) != null && c.noWrap ? (d(), k(l, {
      key: 1,
      componentRef: e.componentRefProps.componentRef,
      componentOptions: e.componentRefProps.componentOptions,
      blockChildren: e.componentRefProps.blockChildren,
      context: e.componentRefProps.context,
      registeredComponents: e.componentRefProps.registeredComponents,
      linkComponent: e.componentRefProps.linkComponent,
      builderBlock: e.componentRefProps.builderBlock,
      includeBlockProps: e.componentRefProps.includeBlockProps,
      isInteractive: e.componentRefProps.isInteractive
    }, null, 8, ["componentRef", "componentOptions", "blockChildren", "context", "registeredComponents", "linkComponent", "builderBlock", "includeBlockProps", "isInteractive"])) : (d(), p(I, { key: 0 }, [
      e.repeatItem ? (d(!0), p(I, { key: 1 }, N(e.repeatItem, (m, C) => (d(), k(h, {
        key: C,
        repeatContext: m.context,
        block: m.block,
        registeredComponents: e.registeredComponents,
        linkComponent: e.linkComponent
      }, null, 8, ["repeatContext", "block", "registeredComponents", "linkComponent"]))), 128)) : (d(), k(u, {
        key: 0,
        Wrapper: e.Tag,
        block: e.processedBlock,
        context: e.context
      }, {
        default: W(() => [
          $(l, {
            componentRef: e.componentRefProps.componentRef,
            componentOptions: e.componentRefProps.componentOptions,
            blockChildren: e.componentRefProps.blockChildren,
            context: e.componentRefProps.context,
            registeredComponents: e.componentRefProps.registeredComponents,
            linkComponent: e.componentRefProps.linkComponent,
            builderBlock: e.componentRefProps.builderBlock,
            includeBlockProps: e.componentRefProps.includeBlockProps,
            isInteractive: e.componentRefProps.isInteractive
          }, null, 8, ["componentRef", "componentOptions", "blockChildren", "context", "registeredComponents", "linkComponent", "builderBlock", "includeBlockProps", "isInteractive"]),
          (d(!0), p(I, null, N(e.childrenWithoutParentComponent, (m, C) => (d(), k(a, {
            key: m.id,
            block: m,
            registeredComponents: e.registeredComponents,
            linkComponent: e.linkComponent,
            context: e.context
          }, null, 8, ["block", "registeredComponents", "linkComponent", "context"]))), 128))
        ]),
        _: 1
      }, 8, ["Wrapper", "block", "context"]))
    ], 64))
  ], 64)) : v("", !0);
}
const be = /* @__PURE__ */ b(tn, [["render", nn]]), Vi = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: be
}, Symbol.toStringTag, { value: "Module" })), on = f({
  name: "builder-blocks-wrapper",
  props: [
    "blocks",
    "parent",
    "path",
    "styleProp",
    "BlocksWrapperProps",
    "BlocksWrapper"
  ],
  computed: {
    className() {
      var e;
      return "builder-blocks" + ((e = this.blocks) != null && e.length ? "" : " no-blocks");
    }
  },
  methods: {
    onClick() {
      var e, t;
      R() && !((e = this.blocks) != null && e.length) && ((t = window.parent) == null || t.postMessage(
        {
          type: "builder.clickEmptyBlocks",
          data: {
            parentElementId: this.parent,
            dataPath: this.path
          }
        },
        "*"
      ));
    },
    onMouseEnter() {
      var e, t;
      R() && !((e = this.blocks) != null && e.length) && ((t = window.parent) == null || t.postMessage(
        {
          type: "builder.hoverEmptyBlocks",
          data: {
            parentElementId: this.parent,
            dataPath: this.path
          }
        },
        "*"
      ));
    }
  }
});
function rn(e, t, n, o, r, i) {
  return d(), k(K(e.BlocksWrapper), S({
    class: e.className + " component-178o76acnws",
    "builder-path": e.path,
    "builder-parent-id": e.parent,
    style: e.styleProp,
    onClick: (s) => e.onClick(),
    onMouseEnter: (s) => e.onMouseEnter(),
    onKeyPress: (s) => e.onClick()
  }, { ...e.BlocksWrapperProps }), {
    default: W(() => [
      E(e.$slots, "default", {}, void 0, !0)
    ]),
    _: 3
  }, 16, ["class", "builder-path", "builder-parent-id", "style", "onClick", "onMouseEnter", "onKeyPress"]);
}
const sn = /* @__PURE__ */ b(on, [["render", rn], ["__scopeId", "data-v-a5b13ef9"]]), an = f({
  name: "builder-blocks",
  components: { BlocksWrapper: sn, Block: be },
  props: [
    "blocks",
    "parent",
    "path",
    "styleProp",
    "context",
    "linkComponent",
    "registeredComponents"
  ],
  inject: {
    builderContext: de.key,
    componentsContext: Ve.key
  }
});
function ln(e, t, n, o, r, i) {
  var a, u;
  const s = g("Block"), l = g("BlocksWrapper");
  return d(), k(l, {
    blocks: e.blocks,
    parent: e.parent,
    path: e.path,
    styleProp: e.styleProp,
    BlocksWrapper: ((a = e.context) == null ? void 0 : a.BlocksWrapper) || e.builderContext.BlocksWrapper,
    BlocksWrapperProps: ((u = e.context) == null ? void 0 : u.BlocksWrapperProps) || e.builderContext.BlocksWrapperProps
  }, {
    default: W(() => [
      e.blocks ? (d(!0), p(I, { key: 0 }, N(e.blocks, (h, c) => (d(), k(s, {
        key: h.id,
        block: h,
        linkComponent: e.linkComponent,
        context: e.context || e.builderContext,
        registeredComponents: e.registeredComponents || e.componentsContext.registeredComponents
      }, null, 8, ["block", "linkComponent", "context", "registeredComponents"]))), 128)) : v("", !0)
    ]),
    _: 1
  }, 8, ["blocks", "parent", "path", "styleProp", "BlocksWrapper", "BlocksWrapperProps"]);
}
const M = /* @__PURE__ */ b(an, [["render", ln]]), cn = f({
  name: "dynamic-div",
  props: []
});
function dn(e, t, n, o, r, i) {
  return d(), p("div", null, [
    E(e.$slots, "default")
  ]);
}
const Q = /* @__PURE__ */ b(cn, [["render", dn]]), un = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: Q
}, Symbol.toStringTag, { value: "Module" })), ge = f({
  render() {
    return Ae("style", {
      innerHTML: this.styles,
      "data-id": this.id
    });
  },
  name: "inlined-styles",
  props: ["styles", "id"]
}), qe = {
  small: {
    min: 320,
    default: 321,
    max: 640
  },
  medium: {
    min: 641,
    default: 642,
    max: 991
  },
  large: {
    min: 990,
    default: 991,
    max: 1200
  }
}, Wi = (e, t = qe) => `@media (max-width: ${t[e].max}px)`, pn = ({
  small: e,
  medium: t
}) => {
  const n = Y(qe);
  if (!e || !t)
    return n;
  const o = Math.floor(e / 2);
  n.small = {
    max: e,
    min: o,
    default: o + 1
  };
  const r = n.small.max + 1;
  n.medium = {
    max: t,
    min: r,
    default: r + 1
  };
  const i = n.medium.max + 1;
  return n.large = {
    max: 2e3,
    // TODO: decide upper limit
    min: i,
    default: i + 1
  }, n;
}, mn = (e) => `builder-columns ${e}-breakpoints`, hn = f({
  name: "builder-columns",
  components: {
    InlinedStyles: ge,
    DynamicRenderer: he,
    Blocks: M,
    DynamicDiv: Q
  },
  props: [
    "space",
    "columns",
    "stackColumnsAt",
    "builderLinkComponent",
    "reverseColumnsWhenStacked",
    "builderContext",
    "builderBlock",
    "builderComponents"
  ],
  data() {
    return {
      gutterSize: typeof this.space == "number" ? this.space || 0 : 20,
      cols: this.columns || [],
      stackAt: this.stackColumnsAt || "tablet",
      flexDir: this.stackColumnsAt === "never" ? "row" : this.reverseColumnsWhenStacked ? "column-reverse" : "column",
      TARGET: A,
      getColumnsClass: mn
    };
  },
  methods: {
    getTagName(e) {
      return e.link ? this.builderLinkComponent || "a" : "div";
    },
    getWidth(e) {
      var t;
      return ((t = this.cols[e]) == null ? void 0 : t.width) || 100 / this.cols.length;
    },
    getColumnCssWidth(e) {
      const t = this.gutterSize * (this.cols.length - 1) / this.cols.length;
      return `calc(${this.getWidth(e)}% - ${t}px)`;
    },
    getTabletStyle({
      stackedStyle: e,
      desktopStyle: t
    }) {
      return this.stackAt === "tablet" ? e : t;
    },
    getMobileStyle({
      stackedStyle: e,
      desktopStyle: t
    }) {
      return this.stackAt === "never" ? t : e;
    },
    columnsCssVars() {
      return {
        "--flex-dir": this.flexDir,
        "--flex-dir-tablet": this.getTabletStyle({
          stackedStyle: this.flexDir,
          desktopStyle: "row"
        })
      };
    },
    columnCssVars(e) {
      const t = e === 0 ? 0 : this.gutterSize, n = this.getColumnCssWidth(e), o = `${t}px`, r = "100%", i = 0;
      return {
        ...{
          display: "flex",
          flexDirection: "column",
          alignItems: "stretch"
        },
        width: n,
        ["margin-left"]: o,
        "--column-width-mobile": this.getMobileStyle({
          stackedStyle: r,
          desktopStyle: n
        }),
        "--column-margin-left-mobile": this.getMobileStyle({
          stackedStyle: i,
          desktopStyle: o
        }),
        "--column-width-tablet": this.getTabletStyle({
          stackedStyle: r,
          desktopStyle: n
        }),
        "--column-margin-left-tablet": this.getTabletStyle({
          stackedStyle: i,
          desktopStyle: o
        })
      };
    },
    getWidthForBreakpointSize(e) {
      var n, o;
      return pn(
        ((o = (n = this.builderContext.content) == null ? void 0 : n.meta) == null ? void 0 : o.breakpoints) || {}
      )[e].max;
    },
    columnsStyles() {
      return `
        @media (max-width: ${this.getWidthForBreakpointSize("medium")}px) {
          .${this.builderBlock.id}-breakpoints {
            flex-direction: var(--flex-dir-tablet);
            align-items: stretch;
          }

          .${this.builderBlock.id}-breakpoints > .builder-column {
            width: var(--column-width-tablet) !important;
            margin-left: var(--column-margin-left-tablet) !important;
          }
        }

        @media (max-width: ${this.getWidthForBreakpointSize("small")}px) {
          .${this.builderBlock.id}-breakpoints {
            flex-direction: var(--flex-dir);
            align-items: stretch;
          }

          .${this.builderBlock.id}-breakpoints > .builder-column {
            width: var(--column-width-mobile) !important;
            margin-left: var(--column-margin-left-mobile) !important;
          }
        },
      `;
    },
    getAttributes(e, t) {
      return {
        ...e.link ? {
          href: e.link
        } : {},
        [ee()]: "builder-column",
        style: Ke(this.columnCssVars(t))
      };
    }
  }
});
function fn(e, t, n, o, r, i) {
  var u;
  const s = g("InlinedStyles"), l = g("Blocks"), a = g("DynamicRenderer");
  return d(), p("div", S({
    class: e.getColumnsClass((u = e.builderBlock) == null ? void 0 : u.id) + " div-2brhoa1k8gd",
    style: e.columnsCssVars()
  }, {}), [
    e.TARGET !== "reactNative" ? (d(), k(s, {
      key: 0,
      id: "builderio-columns",
      styles: e.columnsStyles()
    }, null, 8, ["styles"])) : v("", !0),
    (d(!0), p(I, null, N(e.columns, (h, c) => (d(), k(a, {
      key: c,
      TagName: e.getTagName(h),
      actionAttributes: {},
      attributes: e.getAttributes(h, c)
    }, {
      default: W(() => [
        $(l, {
          path: `component.options.columns.${c}.blocks`,
          parent: e.builderBlock.id,
          styleProp: {
            flexGrow: "1"
          },
          context: e.builderContext,
          registeredComponents: e.builderComponents,
          linkComponent: e.builderLinkComponent,
          blocks: h.blocks
        }, null, 8, ["path", "parent", "context", "registeredComponents", "linkComponent", "blocks"])
      ]),
      _: 2
    }, 1032, ["TagName", "attributes"]))), 128))
  ], 16);
}
const bn = /* @__PURE__ */ b(hn, [["render", fn], ["__scopeId", "data-v-27a4dde0"]]), gn = f({
  name: "builder-fragment-component",
  props: []
});
function yn(e, t, n, o, r, i) {
  return d(), p("span", null, [
    E(e.$slots, "default")
  ]);
}
const kn = /* @__PURE__ */ b(gn, [["render", yn]]);
function we(e) {
  return e.replace(/http(s)?:/, "");
}
function vn(e = "", t, n) {
  const o = new RegExp("([?&])" + t + "=.*?(&|$)", "i"), r = e.indexOf("?") !== -1 ? "&" : "?";
  return e.match(o) ? e.replace(o, "$1" + t + "=" + encodeURIComponent(n) + "$2") : e + r + t + "=" + encodeURIComponent(n);
}
function Cn(e, t) {
  if (!e || !(e != null && e.match(/cdn\.shopify\.com/)) || !t)
    return e;
  if (t === "master")
    return we(e);
  const n = e.match(/(_\d+x(\d+)?)?(\.(jpg|jpeg|gif|png|bmp|bitmap|tiff|tif)(\?v=\d+)?)/i);
  if (n) {
    const o = e.split(n[0]), r = n[3], i = t.match("x") ? t : `${t}x`;
    return we(`${o[0]}_${i}${r}`);
  }
  return null;
}
function ie(e) {
  if (!e)
    return e;
  const t = [100, 200, 400, 800, 1200, 1600, 2e3];
  if (e.match(/builder\.io/)) {
    let n = e;
    const o = Number(e.split("?width=")[1]);
    return isNaN(o) || (n = `${n} ${o}w`), t.filter((r) => r !== o).map((r) => `${vn(e, "width", r)} ${r}w`).concat([n]).join(", ");
  }
  return e.match(/cdn\.shopify\.com/) ? t.map((n) => [Cn(e, `${n}x${n}`), n]).filter(([n]) => !!n).map(([n, o]) => `${n} ${o}w`).concat([e]).join(", ") : e;
}
const Sn = f({
  name: "builder-image",
  props: [
    "image",
    "src",
    "srcset",
    "noWebp",
    "aspectRatio",
    "highPriority",
    "altText",
    "backgroundPosition",
    "backgroundSize",
    "className",
    "sizes",
    "builderBlock",
    "fitContent"
  ],
  computed: {
    srcSetToUse() {
      var n;
      const t = this.image || this.src;
      if (!t || // We can auto add srcset for cdn.builder.io and shopify
      // images, otherwise you can supply this prop manually
      !(t.match(/builder\.io/) || t.match(/cdn\.shopify\.com/)))
        return this.srcset;
      if (this.srcset && ((n = this.image) != null && n.includes("builder.io/api/v1/image"))) {
        if (!this.srcset.includes(this.image.split("?")[0]))
          return console.debug("Removed given srcset"), ie(t);
      } else if (this.image && !this.srcset)
        return ie(t);
      return ie(t);
    },
    webpSrcSet() {
      var e;
      return (e = this.srcSetToUse) != null && e.match(/builder\.io/) && !this.noWebp ? this.srcSetToUse.replace(/\?/g, "?format=webp&") : "";
    },
    aspectRatioCss() {
      const e = {
        position: "absolute",
        height: "100%",
        width: "100%",
        left: "0px",
        top: "0px"
      };
      return this.aspectRatio ? e : void 0;
    }
  }
}), wn = ["srcset"], Tn = ["loading", "fetchpriority", "alt", "role", "src", "srcset", "sizes"], In = {
  key: 2,
  class: "div-1pl23ac79ld-2"
};
function $n(e, t, n, o, r, i) {
  var s, l, a, u, h, c;
  return d(), p(I, null, [
    z("picture", null, [
      e.webpSrcSet ? (d(), p("source", {
        key: 0,
        type: "image/webp",
        srcset: e.webpSrcSet
      }, null, 8, wn)) : v("", !0),
      z("img", {
        loading: e.highPriority ? "eager" : "lazy",
        fetchpriority: e.highPriority ? "high" : "auto",
        alt: e.altText,
        role: e.altText ? void 0 : "presentation",
        style: O({
          objectPosition: e.backgroundPosition || "center",
          objectFit: e.backgroundSize || "cover",
          ...e.aspectRatioCss
        }),
        class: J(
          "builder-image" + (e.className ? " " + e.className : "") + " img-1pl23ac79ld"
        ),
        src: e.image,
        srcset: e.srcSetToUse,
        sizes: e.sizes
      }, null, 14, Tn)
    ]),
    e.aspectRatio && !((l = (s = e.builderBlock) == null ? void 0 : s.children) != null && l.length && e.fitContent) ? (d(), p("div", {
      key: 0,
      class: "builder-image-sizer div-1pl23ac79ld",
      style: O({
        paddingTop: e.aspectRatio * 100 + "%"
      })
    }, null, 4)) : v("", !0),
    (u = (a = e.builderBlock) == null ? void 0 : a.children) != null && u.length && e.fitContent ? E(e.$slots, "default", { key: 1 }, void 0, !0) : v("", !0),
    !e.fitContent && ((c = (h = e.builderBlock) == null ? void 0 : h.children) != null && c.length) ? (d(), p("div", In, [
      E(e.$slots, "default", {}, void 0, !0)
    ])) : v("", !0)
  ], 64);
}
const Rn = /* @__PURE__ */ b(Sn, [["render", $n], ["__scopeId", "data-v-0d528855"]]), Bn = f({
  name: "builder-section-component",
  props: ["attributes", "maxWidth"],
  data() {
    return { filterAttrs: x };
  }
});
function Pn(e, t, n, o, r, i) {
  return d(), p("section", S({
    style: {
      width: "100%",
      alignSelf: "stretch",
      flexGrow: 1,
      boxSizing: "border-box",
      maxWidth: e.maxWidth || 1200,
      display: "flex",
      flexDirection: "column",
      alignItems: "stretch",
      marginLeft: "auto",
      marginRight: "auto"
    }
  }, e.filterAttrs(e.attributes, "v-on:", !1), P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), [
    E(e.$slots, "default")
  ], 16);
}
const En = /* @__PURE__ */ b(Bn, [["render", Pn]]), xn = (e) => {
  if (e === "localhost" || e === "127.0.0.1")
    return e;
  const t = e.split(".");
  return t.length > 2 ? t.slice(1).join(".") : e;
}, ze = ({
  name: e,
  canTrack: t
}) => {
  var n;
  try {
    return t ? (n = document.cookie.split("; ").find((o) => o.startsWith(`${e}=`))) == null ? void 0 : n.split("=")[1] : void 0;
  } catch (o) {
    B.warn("[COOKIE] GET error: ", (o == null ? void 0 : o.message) || o);
    return;
  }
}, Je = async (e) => ze(e), An = (e) => e.map(([t, n]) => n ? `${t}=${n}` : t).filter(D).join("; "), Vn = [["secure", ""], ["SameSite", "None"]], Wn = ({
  name: e,
  value: t,
  expires: n
}) => {
  const r = (T() ? location.protocol === "https:" : !0) ? Vn : [[]], i = n ? [["expires", n.toUTCString()]] : [[]], s = [[e, t], ...i, ["path", "/"], ["domain", xn(window.location.hostname)], ...r];
  return An(s);
}, Ge = async ({
  name: e,
  value: t,
  expires: n,
  canTrack: o
}) => {
  try {
    if (!o)
      return;
    const r = Wn({
      name: e,
      value: t,
      expires: n
    });
    document.cookie = r;
  } catch (r) {
    B.warn("[COOKIE] SET error: ", (r == null ? void 0 : r.message) || r);
  }
}, Nn = "builder.tests", ye = (e) => `${Nn}.${e}`, Fn = ({
  contentId: e
}) => Je({
  name: ye(e),
  canTrack: !0
}), Ln = ({
  contentId: e
}) => ze({
  name: ye(e),
  canTrack: !0
}), On = ({
  contentId: e,
  value: t
}) => Ge({
  name: ye(e),
  value: t,
  canTrack: !0
}), Ye = (e) => D(e.id) && D(e.variations) && Object.keys(e.variations).length > 0, Dn = ({
  id: e,
  variations: t
}) => {
  var r;
  let n = 0;
  const o = Math.random();
  for (const i in t) {
    const s = (r = t[i]) == null ? void 0 : r.testRatio;
    if (n += s, o < n)
      return i;
  }
  return e;
}, Qe = (e) => {
  const t = Dn(e);
  return On({
    contentId: e.id,
    value: t
  }).catch((n) => {
    B.error("could not store A/B test variation: ", n);
  }), t;
}, Xe = ({
  item: e,
  testGroupId: t
}) => {
  const n = e.variations[t];
  return t === e.id || // handle edge-case where `testGroupId` points to non-existing variation
  !n ? {
    testVariationId: e.id,
    testVariationName: "Default"
  } : {
    data: n.data,
    testVariationId: n.id,
    testVariationName: n.name || (n.id === e.id ? "Default" : "")
  };
}, Un = ({
  item: e,
  canTrack: t
}) => {
  if (!t)
    return e;
  if (!e)
    return;
  if (!Ye(e))
    return e;
  const n = Ln({
    contentId: e.id
  }) || Qe({
    variations: e.variations,
    id: e.id
  }), o = Xe({
    item: e,
    testGroupId: n
  });
  return {
    ...e,
    ...o
  };
}, Mn = async ({
  item: e,
  canTrack: t
}) => {
  if (!t || !Ye(e))
    return e;
  const o = await Fn({
    contentId: e.id
  }) || Qe({
    variations: e.variations,
    id: e.id
  }), r = Xe({
    item: e,
    testGroupId: o
  });
  return {
    ...e,
    ...r
  };
}, G = (e) => D(e) ? e : !0, se = {
  "@type": "@builder.io/sdk:Element",
  layerName: "Accordion item title",
  responsiveStyles: {
    large: {
      marginTop: "10px",
      position: "relative",
      display: "flex",
      alignItems: "stretch",
      flexDirection: "column",
      paddingBottom: "10px"
    }
  },
  children: [{
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        textAlign: "left",
        display: "flex",
        flexDirection: "column"
      }
    },
    component: {
      name: "Text",
      options: {
        text: "I am an accordion title. Click me!"
      }
    }
  }]
}, ae = {
  "@type": "@builder.io/sdk:Element",
  layerName: "Accordion item detail",
  responsiveStyles: {
    large: {
      position: "relative",
      display: "flex",
      alignItems: "stretch",
      flexDirection: "column",
      marginTop: "10px",
      paddingBottom: "10px"
    }
  },
  children: [{
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        paddingTop: "50px",
        textAlign: "left",
        display: "flex",
        flexDirection: "column",
        paddingBottom: "50px"
      }
    },
    component: {
      name: "Text",
      options: {
        text: "I am an accordion detail, hello!"
      }
    }
  }]
}, jn = {
  name: "Builder:Accordion",
  canHaveChildren: !0,
  image: "https://cdn.builder.io/api/v1/image/assets%2FagZ9n5CUKRfbL9t6CaJOyVSK4Es2%2Ffab6c1fd3fe542408cbdec078bca7f35",
  defaultStyles: {
    display: "flex",
    flexDirection: "column",
    alignItems: "stretch"
  },
  inputs: [{
    name: "items",
    type: "list",
    broadcast: !0,
    subFields: [{
      name: "title",
      type: "uiBlocks",
      hideFromUI: !0,
      defaultValue: [se]
    }, {
      name: "detail",
      type: "uiBlocks",
      hideFromUI: !0,
      defaultValue: [ae]
    }],
    defaultValue: [{
      title: [se],
      detail: [ae]
    }, {
      title: [se],
      detail: [ae]
    }],
    showIf: (e) => !e.get("useChildrenForItems")
  }, {
    name: "oneAtATime",
    helperText: "Only allow opening one at a time (collapse all others when new item openned)",
    type: "boolean",
    defaultValue: !1
  }, {
    name: "grid",
    helperText: "Display as a grid",
    type: "boolean",
    defaultValue: !1
  }, {
    name: "gridRowWidth",
    helperText: "Display as a grid",
    type: "string",
    showIf: (e) => e.get("grid"),
    defaultValue: "25%"
  }, {
    name: "useChildrenForItems",
    type: "boolean",
    helperText: "Use child elements for each slide, instead of the array. Useful for dynamically repeating items",
    advanced: !0,
    defaultValue: !1,
    onChange: (e) => {
      e.get("useChildrenForItems") === !0 && e.set("items", []);
    }
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !0,
    builderContext: !0,
    builderComponents: !0,
    builderLinkComponent: !0
  }
}, Hn = (e) => e.toString(), Kn = f({
  name: "builder-accordion",
  components: { Blocks: M },
  props: [
    "grid",
    "oneAtATime",
    "items",
    "gridRowWidth",
    "builderBlock",
    "builderContext",
    "builderComponents",
    "builderLinkComponent"
  ],
  data() {
    return { open: [], convertOrderNumberToString: Hn };
  },
  computed: {
    onlyOneAtATime() {
      return !!(this.grid || this.oneAtATime);
    },
    accordionStyles() {
      return {
        display: "flex",
        alignItems: "stretch",
        flexDirection: "column",
        ...this.grid && {
          flexDirection: "row",
          alignItems: "flex-start",
          flexWrap: "wrap"
        }
      };
    },
    accordionTitleStyles() {
      const t = {
        ...{
          display: "flex",
          flexDirection: "column"
        },
        alignItems: "stretch",
        cursor: "pointer"
      };
      return Object.fromEntries(
        Object.entries(t).filter(([n, o]) => o !== void 0)
      );
    },
    openGridItemOrder() {
      let e = null;
      if (this.grid && this.open.length && document) {
        const n = this.open[0];
        let r = document.querySelector(
          `.builder-accordion-title[data-index="${n}"]`
        );
        if (e = n, r) {
          let i = r.getBoundingClientRect();
          for (; r = r && r.nextElementSibling; )
            if (r) {
              if (r.classList.contains("builder-accordion-detail"))
                continue;
              const s = r.getBoundingClientRect();
              if (s.left > i.left) {
                const l = parseInt(
                  r.getAttribute("data-index") || "",
                  10
                );
                isNaN(l) || (i = s, e = l);
              } else
                break;
            }
        }
      }
      return typeof e == "number" && (e = e + 1), e;
    },
    accordionDetailStyles() {
      const e = {
        order: typeof this.openGridItemOrder == "number" ? this.openGridItemOrder : void 0,
        ...this.grid && {
          width: "100%"
        }
      };
      return Object.fromEntries(
        Object.entries(e).filter(([t, n]) => n !== void 0)
      );
    }
  },
  methods: {
    getAccordionTitleClassName(e) {
      return `builder-accordion-title builder-accordion-title-${this.open.includes(e) ? "open" : "closed"}`;
    },
    getAccordionDetailClassName(e) {
      return `builder-accordion-detail builder-accordion-detail-${this.open.includes(e) ? "open" : "closed"}`;
    },
    onClick(e) {
      this.open.includes(e) ? this.open = this.onlyOneAtATime ? [] : this.open.filter((t) => t !== e) : this.open = this.onlyOneAtATime ? [e] : this.open.concat(e);
    }
  }
}), qn = ["data-index", "onClick"];
function zn(e, t, n, o, r, i) {
  const s = g("Blocks");
  return d(), p("div", {
    class: "builder-accordion",
    style: O(e.accordionStyles)
  }, [
    (d(!0), p(I, null, N(e.items, (l, a) => (d(), p(I, { key: a }, [
      z("div", {
        class: J(e.getAccordionTitleClassName(a)),
        style: O({
          ...e.accordionTitleStyles,
          width: e.grid ? e.gridRowWidth : void 0,
          order: e.openGridItemOrder !== null ? e.convertOrderNumberToString(a) : e.convertOrderNumberToString(a + 1)
        }),
        "data-index": a,
        onClick: (u) => e.onClick(a)
      }, [
        $(s, {
          blocks: l.title,
          path: `items.${a}.title`,
          parent: e.builderBlock.id,
          context: e.builderContext,
          registeredComponents: e.builderComponents,
          linkComponent: e.builderLinkComponent
        }, null, 8, ["blocks", "path", "parent", "context", "registeredComponents", "linkComponent"])
      ], 14, qn),
      e.open.includes(a) ? (d(), p("div", {
        key: 0,
        class: J(e.getAccordionDetailClassName(a)),
        style: O(e.accordionDetailStyles)
      }, [
        $(s, {
          blocks: l.detail,
          path: `items.${a}.detail`,
          parent: e.builderBlock.id,
          context: e.builderContext,
          registeredComponents: e.builderComponents,
          linkComponent: e.builderLinkComponent
        }, null, 8, ["blocks", "path", "parent", "context", "registeredComponents", "linkComponent"])
      ], 6)) : v("", !0)
    ], 64))), 128))
  ], 4);
}
const Jn = /* @__PURE__ */ b(Kn, [["render", zn]]), Gn = {
  name: "Core:Button",
  image: "https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2F81a15681c3e74df09677dfc57a615b13",
  defaultStyles: {
    // TODO: make min width more intuitive and set one
    appearance: "none",
    paddingTop: "15px",
    paddingBottom: "15px",
    paddingLeft: "25px",
    paddingRight: "25px",
    backgroundColor: "#000000",
    color: "white",
    borderRadius: "4px",
    textAlign: "center",
    cursor: "pointer"
  },
  inputs: [{
    name: "text",
    type: "text",
    defaultValue: "Click me!",
    bubble: !0
  }, {
    name: "link",
    type: "url",
    bubble: !0
  }, {
    name: "openLinkInNewTab",
    type: "boolean",
    defaultValue: !1,
    friendlyName: "Open link in new tab"
  }],
  static: !0,
  noWrap: !0,
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !0
  }
}, Yn = {
  // TODO: ways to statically preprocess JSON for references, functions, etc
  name: "Columns",
  isRSC: !0,
  inputs: [{
    name: "columns",
    type: "array",
    broadcast: !0,
    subFields: [{
      name: "blocks",
      type: "array",
      hideFromUI: !0,
      defaultValue: [{
        "@type": "@builder.io/sdk:Element",
        responsiveStyles: {
          large: {
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
            flexShrink: "0",
            position: "relative",
            marginTop: "30px",
            textAlign: "center",
            lineHeight: "normal",
            height: "auto",
            minHeight: "20px",
            minWidth: "20px",
            overflow: "hidden"
          }
        },
        component: {
          name: "Image",
          options: {
            image: "https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d",
            backgroundPosition: "center",
            backgroundSize: "cover",
            aspectRatio: 0.7004048582995948
          }
        }
      }, {
        "@type": "@builder.io/sdk:Element",
        responsiveStyles: {
          large: {
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
            flexShrink: "0",
            position: "relative",
            marginTop: "30px",
            textAlign: "center",
            lineHeight: "normal",
            height: "auto"
          }
        },
        component: {
          name: "Text",
          options: {
            text: "<p>Enter some text...</p>"
          }
        }
      }]
    }, {
      name: "width",
      type: "number",
      hideFromUI: !0,
      helperText: "Width %, e.g. set to 50 to fill half of the space"
    }, {
      name: "link",
      type: "url",
      helperText: "Optionally set a url that clicking this column will link to"
    }],
    defaultValue: [{
      blocks: [{
        "@type": "@builder.io/sdk:Element",
        responsiveStyles: {
          large: {
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
            flexShrink: "0",
            position: "relative",
            marginTop: "30px",
            textAlign: "center",
            lineHeight: "normal",
            height: "auto",
            minHeight: "20px",
            minWidth: "20px",
            overflow: "hidden"
          }
        },
        component: {
          name: "Image",
          options: {
            image: "https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d",
            backgroundPosition: "center",
            backgroundSize: "cover",
            aspectRatio: 0.7004048582995948
          }
        }
      }, {
        "@type": "@builder.io/sdk:Element",
        responsiveStyles: {
          large: {
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
            flexShrink: "0",
            position: "relative",
            marginTop: "30px",
            textAlign: "center",
            lineHeight: "normal",
            height: "auto"
          }
        },
        component: {
          name: "Text",
          options: {
            text: "<p>Enter some text...</p>"
          }
        }
      }]
    }, {
      blocks: [{
        "@type": "@builder.io/sdk:Element",
        responsiveStyles: {
          large: {
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
            flexShrink: "0",
            position: "relative",
            marginTop: "30px",
            textAlign: "center",
            lineHeight: "normal",
            height: "auto",
            minHeight: "20px",
            minWidth: "20px",
            overflow: "hidden"
          }
        },
        component: {
          name: "Image",
          options: {
            image: "https://builder.io/api/v1/image/assets%2Fpwgjf0RoYWbdnJSbpBAjXNRMe9F2%2Ffb27a7c790324294af8be1c35fe30f4d",
            backgroundPosition: "center",
            backgroundSize: "cover",
            aspectRatio: 0.7004048582995948
          }
        }
      }, {
        "@type": "@builder.io/sdk:Element",
        responsiveStyles: {
          large: {
            display: "flex",
            flexDirection: "column",
            alignItems: "stretch",
            flexShrink: "0",
            position: "relative",
            marginTop: "30px",
            textAlign: "center",
            lineHeight: "normal",
            height: "auto"
          }
        },
        component: {
          name: "Text",
          options: {
            text: "<p>Enter some text...</p>"
          }
        }
      }]
    }],
    onChange: (e) => {
      function t() {
        n.forEach((o) => {
          o.delete("width");
        });
      }
      const n = e.get("columns");
      Array.isArray(n) && n.find((r) => r.get("width")) && (n.find((i) => !i.get("width")) || n.reduce((l, a) => l + a.get("width"), 0) !== 100) && t();
    }
  }, {
    name: "space",
    type: "number",
    defaultValue: 20,
    helperText: "Size of gap between columns",
    advanced: !0
  }, {
    name: "stackColumnsAt",
    type: "string",
    defaultValue: "tablet",
    helperText: "Convert horizontal columns to vertical at what device size",
    enum: ["tablet", "mobile", "never"],
    advanced: !0
  }, {
    name: "reverseColumnsWhenStacked",
    type: "boolean",
    defaultValue: !1,
    helperText: "When stacking columns for mobile devices, reverse the ordering",
    advanced: !0
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !0,
    builderContext: !0,
    builderComponents: !0,
    builderLinkComponent: !0
  }
}, Qn = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Fragment",
  static: !0,
  hidden: !0,
  canHaveChildren: !0,
  noWrap: !0
}, Xn = {
  name: "Image",
  static: !0,
  image: "https://firebasestorage.googleapis.com/v0/b/builder-3b0a2.appspot.com/o/images%2Fbaseline-insert_photo-24px.svg?alt=media&token=4e5d0ef4-f5e8-4e57-b3a9-38d63a9b9dc4",
  defaultStyles: {
    position: "relative",
    minHeight: "20px",
    minWidth: "20px",
    overflow: "hidden"
  },
  canHaveChildren: !0,
  inputs: [{
    name: "image",
    type: "file",
    bubble: !0,
    allowedFileTypes: ["jpeg", "jpg", "png", "svg", "webp"],
    required: !0,
    defaultValue: "https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2F72c80f114dc149019051b6852a9e3b7a",
    onChange: (e) => {
      e.delete("srcset"), e.delete("noWebp");
      function n(s, l = 6e4) {
        return new Promise((a, u) => {
          const h = document.createElement("img");
          let c = !1;
          h.onload = () => {
            c = !0, a(h);
          }, h.addEventListener("error", (m) => {
            console.warn("Image load failed", m.error), u(m.error);
          }), h.src = s, setTimeout(() => {
            c || u(new Error("Image load timed out"));
          }, l);
        });
      }
      function o(s) {
        return Math.round(s * 1e3) / 1e3;
      }
      const r = e.get("image"), i = e.get("aspectRatio");
      if (fetch(r).then((s) => s.blob()).then((s) => {
        s.type.includes("svg") && e.set("noWebp", !0);
      }), r && (!i || i === 0.7041))
        return n(r).then((s) => {
          const l = e.get("aspectRatio");
          e.get("image") === r && (!l || l === 0.7041) && s.width && s.height && (e.set("aspectRatio", o(s.height / s.width)), e.set("height", s.height), e.set("width", s.width));
        });
    }
  }, {
    name: "backgroundSize",
    type: "text",
    defaultValue: "cover",
    enum: [{
      label: "contain",
      value: "contain",
      helperText: "The image should never get cropped"
    }, {
      label: "cover",
      value: "cover",
      helperText: "The image should fill it's box, cropping when needed"
    }]
  }, {
    name: "backgroundPosition",
    type: "text",
    defaultValue: "center",
    enum: ["center", "top", "left", "right", "bottom", "top left", "top right", "bottom left", "bottom right"]
  }, {
    name: "altText",
    type: "string",
    helperText: "Text to display when the user has images off"
  }, {
    name: "height",
    type: "number",
    hideFromUI: !0
  }, {
    name: "width",
    type: "number",
    hideFromUI: !0
  }, {
    name: "sizes",
    type: "string",
    hideFromUI: !0
  }, {
    name: "srcset",
    type: "string",
    hideFromUI: !0
  }, {
    name: "lazy",
    type: "boolean",
    defaultValue: !0,
    hideFromUI: !0
  }, {
    name: "fitContent",
    type: "boolean",
    helperText: "When child blocks are provided, fit to them instead of using the image's aspect ratio",
    defaultValue: !0
  }, {
    name: "highPriority",
    type: "boolean",
    advanced: !0,
    helperText: "Mark this image as high priority compared to other images on the page. This prevents lazy loading of the image and tells the browser to load this image before others on the page."
  }, {
    name: "aspectRatio",
    type: "number",
    helperText: "This is the ratio of height/width, e.g. set to 1.5 for a 300px wide and 200px tall photo. Set to 0 to not force the image to maintain it's aspect ratio",
    advanced: !0,
    defaultValue: 0.7041
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !0,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  }
}, Zn = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Core:Section",
  static: !0,
  image: "https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2F682efef23ace49afac61748dd305c70a",
  inputs: [{
    name: "maxWidth",
    type: "number",
    defaultValue: 1200
  }, {
    name: "lazyLoad",
    type: "boolean",
    defaultValue: !1,
    advanced: !0,
    description: "Only render this section when in view"
  }],
  defaultStyles: {
    paddingLeft: "20px",
    paddingRight: "20px",
    paddingTop: "50px",
    paddingBottom: "50px",
    marginTop: "0px",
    width: "100vw",
    marginLeft: "calc(50% - 50vw)"
  },
  canHaveChildren: !0,
  defaultChildren: [{
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        textAlign: "center"
      }
    },
    component: {
      name: "Text",
      options: {
        text: "<p><b>I am a section! My content keeps from getting too wide, so that it's easy to read even on big screens.</b></p><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur</p>"
      }
    }
  }]
}, _n = {
  name: "Slot",
  isRSC: !0,
  description: "Allow child blocks to be inserted into this content when used as a Symbol",
  docsLink: "https://www.builder.io/c/docs/symbols-with-blocks",
  image: "https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2F3aad6de36eae43b59b52c85190fdef56",
  // Maybe wrap this for canHaveChildren so bind children to this hm
  inputs: [{
    name: "name",
    type: "string",
    required: !0,
    defaultValue: "children"
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !0,
    builderComponents: !1,
    builderLinkComponent: !1
  }
}, eo = f({
  name: "builder-slot",
  components: { Blocks: M },
  props: ["builderContext", "name"]
});
function to(e, t, n, o, r, i) {
  var l, a, u;
  const s = g("Blocks");
  return d(), p("div", S(
    { style: {
      pointerEvents: "auto"
    } },
    !((l = e.builderContext.context) != null && l.symbolId) && {
      "builder-slot": e.name
    }
  ), [
    $(s, {
      parent: (a = e.builderContext.context) == null ? void 0 : a.symbolId,
      path: `symbol.data.${e.name}`,
      context: e.builderContext,
      blocks: (u = e.builderContext.rootState) == null ? void 0 : u[e.name]
    }, null, 8, ["parent", "path", "context", "blocks"])
  ], 16);
}
const no = /* @__PURE__ */ b(eo, [["render", to]]), oo = {
  name: "Symbol",
  noWrap: !0,
  static: !0,
  isRSC: !0,
  inputs: [{
    name: "symbol",
    type: "uiSymbol"
  }, {
    name: "dataOnly",
    helperText: "Make this a data symbol that doesn't display any UI",
    type: "boolean",
    defaultValue: !1,
    advanced: !0,
    hideFromUI: !0
  }, {
    name: "inheritState",
    helperText: "Inherit the parent component state and data",
    type: "boolean",
    defaultValue: !1,
    advanced: !0
  }, {
    name: "renderToLiquid",
    helperText: "Render this symbols contents to liquid. Turn off to fetch with javascript and use custom targeting",
    type: "boolean",
    defaultValue: !1,
    advanced: !0,
    hideFromUI: !0
  }, {
    name: "useChildren",
    hideFromUI: !0,
    type: "boolean"
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !0,
    builderContext: !0,
    builderComponents: !0,
    builderLinkComponent: !0
  }
}, le = {
  "@type": "@builder.io/sdk:Element",
  responsiveStyles: {
    large: {
      paddingLeft: "20px",
      paddingRight: "20px",
      paddingTop: "10px",
      paddingBottom: "10px",
      minWidth: "100px",
      textAlign: "center",
      display: "flex",
      flexDirection: "column",
      cursor: "pointer",
      userSelect: "none"
    }
  },
  component: {
    name: "Text",
    options: {
      text: "New tab"
    }
  }
}, ce = {
  "@type": "@builder.io/sdk:Element",
  responsiveStyles: {
    large: {
      height: "200px",
      display: "flex",
      marginTop: "20px",
      flexDirection: "column"
    }
  },
  component: {
    name: "Text",
    options: {
      text: "New tab content "
    }
  }
}, ro = {
  name: "Builder: Tabs",
  inputs: [{
    name: "tabs",
    type: "list",
    broadcast: !0,
    subFields: [{
      name: "label",
      type: "uiBlocks",
      hideFromUI: !0,
      defaultValue: [le]
    }, {
      name: "content",
      type: "uiBlocks",
      hideFromUI: !0,
      defaultValue: [ce]
    }],
    defaultValue: [{
      label: [{
        ...le,
        component: {
          name: "Text",
          options: {
            text: "Tab 1"
          }
        }
      }],
      content: [{
        ...ce,
        component: {
          name: "Text",
          options: {
            text: "Tab 1 content"
          }
        }
      }]
    }, {
      label: [{
        ...le,
        component: {
          name: "Text",
          options: {
            text: "Tab 2"
          }
        }
      }],
      content: [{
        ...ce,
        component: {
          name: "Text",
          options: {
            text: "Tab 2 content"
          }
        }
      }]
    }]
  }, {
    name: "activeTabStyle",
    type: "uiStyle",
    helperText: "CSS styles for the active tab",
    defaultValue: {
      backgroundColor: "rgba(0, 0, 0, 0.1)"
    }
  }, {
    name: "defaultActiveTab",
    type: "number",
    helperText: 'Default tab to open to. Set to "1" for the first tab, "2" for the second, or choose "0" for none',
    defaultValue: 1,
    advanced: !0
  }, {
    name: "collapsible",
    type: "boolean",
    helperText: "If on, clicking an open tab closes it so no tabs are active",
    defaultValue: !1,
    advanced: !0
  }, {
    name: "tabHeaderLayout",
    type: "enum",
    helperText: "Change the layout of the tab headers (uses justify-content)",
    defaultValue: "flex-start",
    enum: [{
      label: "Center",
      value: "center"
    }, {
      label: "Space between",
      value: "space-between"
    }, {
      label: "Space around",
      value: "space-around"
    }, {
      label: "Left",
      value: "flex-start"
    }, {
      label: "Right",
      value: "flex-end"
    }]
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !0,
    builderContext: !0,
    builderComponents: !0,
    builderLinkComponent: !0
  }
}, io = f({
  name: "builder-tabs",
  components: { Blocks: M },
  props: [
    "defaultActiveTab",
    "tabs",
    "collapsible",
    "tabHeaderLayout",
    "activeTabStyle",
    "builderBlock",
    "builderContext",
    "builderComponents",
    "builderLinkComponent"
  ],
  data() {
    return { activeTab: this.defaultActiveTab ? this.defaultActiveTab - 1 : 0 };
  },
  methods: {
    activeTabContent(e) {
      return this.tabs && this.tabs[e].content;
    },
    onClick(e) {
      e === this.activeTab && this.collapsible ? this.activeTab = -1 : this.activeTab = e;
    }
  }
}), so = ["onClick"], ao = { key: 0 };
function lo(e, t, n, o, r, i) {
  const s = g("Blocks");
  return d(), p("div", null, [
    z("div", {
      class: "builder-tabs-wrap",
      style: O({
        display: "flex",
        flexDirection: "row",
        justifyContent: e.tabHeaderLayout || "flex-start",
        overflow: "auto"
      })
    }, [
      (d(!0), p(I, null, N(e.tabs, (l, a) => (d(), p("span", {
        key: a,
        class: J(`builder-tab-wrap ${e.activeTab === a ? "builder-tab-active" : ""}`),
        style: O({
          ...e.activeTab === a ? e.activeTabStyle : {}
        }),
        onClick: (u) => e.onClick(a)
      }, [
        $(s, {
          parent: e.builderBlock.id,
          path: `component.options.tabs.${a}.label`,
          blocks: l.label,
          context: e.builderContext,
          registeredComponents: e.builderComponents,
          linkComponent: e.builderLinkComponent
        }, null, 8, ["parent", "path", "blocks", "context", "registeredComponents", "linkComponent"])
      ], 14, so))), 128))
    ], 4),
    e.activeTabContent(e.activeTab) ? (d(), p("div", ao, [
      $(s, {
        parent: e.builderBlock.id,
        path: `component.options.tabs.${e.activeTab}.content`,
        blocks: e.activeTabContent(e.activeTab),
        context: e.builderContext,
        registeredComponents: e.builderComponents,
        linkComponent: e.builderLinkComponent
      }, null, 8, ["parent", "path", "blocks", "context", "registeredComponents", "linkComponent"])
    ])) : v("", !0)
  ]);
}
const co = /* @__PURE__ */ b(io, [["render", lo]]), uo = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Text",
  static: !0,
  isRSC: !0,
  image: "https://firebasestorage.googleapis.com/v0/b/builder-3b0a2.appspot.com/o/images%2Fbaseline-text_fields-24px%20(1).svg?alt=media&token=12177b73-0ee3-42ca-98c6-0dd003de1929",
  inputs: [{
    name: "text",
    type: "html",
    required: !0,
    autoFocus: !0,
    bubble: !0,
    defaultValue: "Enter some text..."
  }],
  defaultStyles: {
    lineHeight: "normal",
    height: "auto",
    textAlign: "center"
  }
}, po = f({
  name: "builder-text",
  props: ["text"]
}), mo = ["innerHTML"];
function ho(e, t, n, o, r, i) {
  var s;
  return d(), p("div", {
    class: "builder-text",
    innerHTML: ((s = e.text) == null ? void 0 : s.toString()) || "",
    style: {
      outline: "none"
    }
  }, null, 8, mo);
}
const fo = /* @__PURE__ */ b(po, [["render", ho]]), bo = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Custom Code",
  static: !0,
  requiredPermissions: ["editCode"],
  inputs: [{
    name: "code",
    type: "html",
    required: !0,
    defaultValue: "<p>Hello there, I am custom HTML code!</p>",
    code: !0
  }, {
    name: "replaceNodes",
    type: "boolean",
    helperText: "Preserve server rendered dom nodes",
    advanced: !0
  }, {
    name: "scriptsClientOnly",
    type: "boolean",
    defaultValue: !1,
    helperText: "Only print and run scripts on the client. Important when scripts influence DOM that could be replaced when client loads",
    advanced: !0
  }]
}, go = f({
  name: "builder-custom-code",
  props: ["replaceNodes", "code"],
  data() {
    return { scriptsInserted: [], scriptsRun: [] };
  },
  mounted() {
    var t;
    if (!((t = this.$refs.elementRef) != null && t.getElementsByTagName) || typeof window > "u")
      return;
    const e = this.$refs.elementRef.getElementsByTagName("script");
    for (let n = 0; n < e.length; n++) {
      const o = e[n];
      if (o.src) {
        if (this.scriptsInserted.includes(o.src))
          continue;
        this.scriptsInserted.push(o.src);
        const r = document.createElement("script");
        r.async = !0, r.src = o.src, document.head.appendChild(r);
      } else if (!o.type || [
        "text/javascript",
        "application/javascript",
        "application/ecmascript"
      ].includes(o.type)) {
        if (this.scriptsRun.includes(o.innerText))
          continue;
        try {
          this.scriptsRun.push(o.innerText), new Function(o.innerText)();
        } catch (r) {
          console.warn("`CustomCode`: Error running script:", r);
        }
      }
    }
  }
}), yo = ["innerHTML"];
function ko(e, t, n, o, r, i) {
  return d(), p("div", {
    ref: "elementRef",
    class: J("builder-custom-code" + (e.replaceNodes ? " replace-nodes" : "")),
    innerHTML: e.code
  }, null, 10, yo);
}
const vo = /* @__PURE__ */ b(go, [["render", ko]]), Co = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Embed",
  static: !0,
  inputs: [{
    name: "url",
    type: "url",
    required: !0,
    defaultValue: "",
    helperText: "e.g. enter a youtube url, google map, etc"
  }, {
    name: "content",
    type: "html",
    defaultValue: '<div style="padding: 20px; text-align: center">(Choose an embed URL)<div>',
    hideFromUI: !0
  }]
}, So = ["text/javascript", "application/javascript", "application/ecmascript"], wo = (e) => So.includes(e.type), To = f({
  name: "builder-embed",
  props: ["content"],
  data() {
    return { scriptsInserted: [], scriptsRun: [], ranInitFn: !1 };
  },
  watch: {
    onUpdateHook0: {
      handler() {
        this.$refs.elem && !this.ranInitFn && (this.ranInitFn = !0, this.findAndRunScripts());
      },
      immediate: !0
    }
  },
  computed: {
    onUpdateHook0() {
      return {
        0: this.$refs.elem,
        1: this.ranInitFn
      };
    }
  },
  methods: {
    findAndRunScripts() {
      if (!this.$refs.elem || !this.$refs.elem.getElementsByTagName)
        return;
      const e = this.$refs.elem.getElementsByTagName("script");
      for (let t = 0; t < e.length; t++) {
        const n = e[t];
        if (n.src && !this.scriptsInserted.includes(n.src)) {
          this.scriptsInserted.push(n.src);
          const o = document.createElement("script");
          o.async = !0, o.src = n.src, document.head.appendChild(o);
        } else if (wo(n) && !this.scriptsRun.includes(n.innerText))
          try {
            this.scriptsRun.push(n.innerText), new Function(n.innerText)();
          } catch (o) {
            console.warn("`Embed`: Error running script:", o);
          }
      }
    }
  }
}), Io = ["innerHTML"];
function $o(e, t, n, o, r, i) {
  return d(), p("div", {
    class: "builder-embed",
    ref: "elem",
    innerHTML: e.content
  }, null, 8, Io);
}
const Ro = /* @__PURE__ */ b(To, [["render", $o]]), Bo = {
  name: "Form:Form",
  // editableTags: ['builder-form-error']
  defaults: {
    responsiveStyles: {
      large: {
        marginTop: "15px",
        paddingBottom: "15px"
      }
    }
  },
  image: "https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2Fef36d2a846134910b64b88e6d18c5ca5",
  inputs: [{
    name: "sendSubmissionsTo",
    type: "string",
    // TODO: save to builder data and user can download as csv
    // TODO: easy for mode too or computed add/remove fields form mode
    // so you can edit details and high level mode at same time...
    // Later - more integrations like mailchimp
    // /api/v1/form-submit?to=mailchimp
    enum: [{
      label: "Send to email",
      value: "email",
      helperText: "Send form submissions to the email address of your choosing"
    }, {
      label: "Custom",
      value: "custom",
      helperText: "Handle where the form requests go manually with a little code, e.g. to your own custom backend"
    }],
    defaultValue: "email"
  }, {
    name: "sendSubmissionsToEmail",
    type: "string",
    required: !0,
    // TODO: required: () => options.get("sendSubmissionsTo") === "email"
    defaultValue: "your@email.com",
    showIf: 'options.get("sendSubmissionsTo") === "email"'
  }, {
    name: "sendWithJs",
    type: "boolean",
    helperText: "Set to false to use basic html form action",
    defaultValue: !0,
    showIf: 'options.get("sendSubmissionsTo") === "custom"'
  }, {
    name: "name",
    type: "string",
    defaultValue: "My form"
    // advanced: true
  }, {
    name: "action",
    type: "string",
    helperText: "URL to send the form data to",
    showIf: 'options.get("sendSubmissionsTo") === "custom"'
  }, {
    name: "contentType",
    type: "string",
    defaultValue: "application/json",
    advanced: !0,
    // TODO: do automatically if file input
    enum: ["application/json", "multipart/form-data", "application/x-www-form-urlencoded"],
    showIf: 'options.get("sendSubmissionsTo") === "custom" && options.get("sendWithJs") === true'
  }, {
    name: "method",
    type: "string",
    showIf: 'options.get("sendSubmissionsTo") === "custom"',
    defaultValue: "POST",
    advanced: !0
  }, {
    name: "previewState",
    type: "string",
    // TODO: persist: false flag
    enum: ["unsubmitted", "sending", "success", "error"],
    defaultValue: "unsubmitted",
    helperText: 'Choose a state to edit, e.g. choose "success" to show what users see on success and edit the message',
    showIf: 'options.get("sendSubmissionsTo") !== "zapier" && options.get("sendWithJs") === true'
  }, {
    name: "successUrl",
    type: "url",
    helperText: "Optional URL to redirect the user to on form submission success",
    showIf: 'options.get("sendSubmissionsTo") !== "zapier" && options.get("sendWithJs") === true'
  }, {
    name: "resetFormOnSubmit",
    type: "boolean",
    showIf: "options.get('sendSubmissionsTo') === 'custom' && options.get('sendWithJs') === true",
    advanced: !0
  }, {
    name: "successMessage",
    type: "uiBlocks",
    hideFromUI: !0,
    defaultValue: [{
      "@type": "@builder.io/sdk:Element",
      responsiveStyles: {
        large: {
          marginTop: "10px"
        }
      },
      component: {
        name: "Text",
        options: {
          text: "<span>Thanks!</span>"
        }
      }
    }]
  }, {
    name: "validate",
    type: "boolean",
    defaultValue: !0,
    advanced: !0
  }, {
    name: "errorMessagePath",
    type: "text",
    advanced: !0,
    helperText: 'Path to where to get the error message from in a JSON response to display to the user, e.g. "error.message" for a response like { "error": { "message": "this username is taken" }}'
  }, {
    name: "errorMessage",
    type: "uiBlocks",
    hideFromUI: !0,
    defaultValue: [{
      "@type": "@builder.io/sdk:Element",
      responsiveStyles: {
        large: {
          marginTop: "10px"
        }
      },
      bindings: {
        "component.options.text": "state.formErrorMessage || block.component.options.text"
      },
      component: {
        name: "Text",
        options: {
          text: "<span>Form submission error :( Please check your answers and try again</span>"
        }
      }
    }]
  }, {
    name: "sendingMessage",
    type: "uiBlocks",
    hideFromUI: !0,
    defaultValue: [{
      "@type": "@builder.io/sdk:Element",
      responsiveStyles: {
        large: {
          marginTop: "10px"
        }
      },
      component: {
        name: "Text",
        options: {
          text: "<span>Sending...</span>"
        }
      }
    }]
  }, {
    name: "customHeaders",
    type: "map",
    valueType: {
      type: "string"
    },
    advanced: !0,
    showIf: 'options.get("sendSubmissionsTo") === "custom" && options.get("sendWithJs") === true'
  }],
  noWrap: !0,
  canHaveChildren: !0,
  defaultChildren: [{
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        marginTop: "10px"
      }
    },
    component: {
      name: "Text",
      options: {
        text: "<span>Enter your name</span>"
      }
    }
  }, {
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        marginTop: "10px"
      }
    },
    component: {
      name: "Form:Input",
      options: {
        name: "name",
        placeholder: "Jane Doe"
      }
    }
  }, {
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        marginTop: "10px"
      }
    },
    component: {
      name: "Text",
      options: {
        text: "<span>Enter your email</span>"
      }
    }
  }, {
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        marginTop: "10px"
      }
    },
    component: {
      name: "Form:Input",
      options: {
        name: "email",
        placeholder: "jane@doe.com"
      }
    }
  }, {
    "@type": "@builder.io/sdk:Element",
    responsiveStyles: {
      large: {
        marginTop: "10px"
      }
    },
    component: {
      name: "Form:SubmitButton",
      options: {
        text: "Submit"
      }
    }
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !0,
    builderContext: !0,
    builderComponents: !0,
    builderLinkComponent: !0
  }
}, Po = ["production", "qa", "test", "development", "dev", "cdn-qa", "cloud", "fast", "cdn2", "cdn-prod"], Eo = () => {
  const e = process.env.NODE_ENV || "production";
  return Po.includes(e) ? e : "production";
}, xo = (e, t, n) => {
  const o = String.prototype.split.call(t, /[,[\].]+?/).filter(Boolean).reduce((r, i) => r != null ? r[i] : r, e);
  return o === void 0 || o === e ? n : o;
}, Ao = f({
  name: "builder-form-component",
  components: { Block: be, Blocks: M },
  props: [
    "builderContext",
    "previewState",
    "sendWithJs",
    "sendSubmissionsTo",
    "action",
    "customHeaders",
    "contentType",
    "sendSubmissionsToEmail",
    "name",
    "method",
    "errorMessagePath",
    "resetFormOnSubmit",
    "successUrl",
    "validate",
    "attributes",
    "builderBlock",
    "builderComponents",
    "builderLinkComponent",
    "errorMessage",
    "sendingMessage",
    "successMessage"
  ],
  data() {
    return {
      formState: "unsubmitted",
      responseData: null,
      formErrorMessage: "",
      filterAttrs: x
    };
  },
  methods: {
    mergeNewRootState(e) {
      var n, o;
      const t = {
        ...this.builderContext.rootState,
        ...e
      };
      this.builderContext.rootSetState ? (o = (n = this.builderContext).rootSetState) == null || o.call(n, t) : this.builderContext.rootState = t;
    },
    submissionState() {
      return R() && this.previewState || this.formState;
    },
    onSubmit(e) {
      var n;
      const t = this.sendWithJs || this.sendSubmissionsTo === "email";
      if (this.sendSubmissionsTo === "zapier")
        e.preventDefault();
      else if (t) {
        if (!(this.action || this.sendSubmissionsTo === "email")) {
          e.preventDefault();
          return;
        }
        e.preventDefault();
        const o = e.currentTarget || e.target, r = this.customHeaders || {};
        let i;
        const s = new FormData(o), l = Array.from(o.querySelectorAll("input,select,textarea")).filter((c) => !!c.name).map((c) => {
          let m;
          const C = c.name;
          if (c instanceof HTMLInputElement)
            if (c.type === "radio") {
              if (c.checked)
                return m = c.name, {
                  key: C,
                  value: m
                };
            } else if (c.type === "checkbox")
              m = c.checked;
            else if (c.type === "number" || c.type === "range") {
              const y = c.valueAsNumber;
              isNaN(y) || (m = y);
            } else
              c.type === "file" ? m = c.files : m = c.value;
          else
            m = c.value;
          return {
            key: C,
            value: m
          };
        });
        let a = this.contentType;
        if (this.sendSubmissionsTo === "email" && (a = "multipart/form-data"), Array.from(l).forEach(({ value: c }) => {
          (c instanceof File || Array.isArray(c) && c[0] instanceof File || c instanceof FileList) && (a = "multipart/form-data");
        }), a !== "application/json")
          i = s;
        else {
          const c = {};
          Array.from(l).forEach(({ value: m, key: C }) => {
            Oe(c, C, m);
          }), i = JSON.stringify(c);
        }
        a && a !== "multipart/form-data" && (t && ((n = this.action) != null && n.includes("zapier.com")) || (r["content-type"] = a));
        const u = new CustomEvent("presubmit", {
          detail: {
            body: i
          }
        });
        if (this.$refs.formRef && (this.$refs.formRef.dispatchEvent(u), u.defaultPrevented))
          return;
        this.formState = "sending";
        const h = `${Eo() === "dev" ? "http://localhost:5000" : "https://builder.io"}/api/v1/form-submit?apiKey=${this.builderContext.apiKey}&to=${btoa(
          this.sendSubmissionsToEmail || ""
        )}&name=${encodeURIComponent(this.name || "")}`;
        fetch(
          this.sendSubmissionsTo === "email" ? h : this.action,
          {
            body: i,
            headers: r,
            method: this.method || "post"
          }
        ).then(
          async (c) => {
            let m;
            const C = c.headers.get("content-type");
            if (C && C.indexOf("application/json") !== -1 ? m = await c.json() : m = await c.text(), !c.ok && this.errorMessagePath) {
              let y = xo(m, this.errorMessagePath);
              y && (typeof y != "string" && (y = JSON.stringify(y)), this.formErrorMessage = y, this.mergeNewRootState({
                formErrorMessage: y
              }));
            }
            if (this.responseData = m, this.formState = c.ok ? "success" : "error", c.ok) {
              const y = new CustomEvent("submit:success", {
                detail: {
                  res: c,
                  body: m
                }
              });
              if (this.$refs.formRef) {
                if (this.$refs.formRef.dispatchEvent(y), y.defaultPrevented)
                  return;
                this.resetFormOnSubmit !== !1 && this.$refs.formRef.reset();
              }
              if (this.successUrl)
                if (this.$refs.formRef) {
                  const F = new CustomEvent("route", {
                    detail: {
                      url: this.successUrl
                    }
                  });
                  this.$refs.formRef.dispatchEvent(F), F.defaultPrevented || (location.href = this.successUrl);
                } else
                  location.href = this.successUrl;
            }
          },
          (c) => {
            const m = new CustomEvent("submit:error", {
              detail: {
                error: c
              }
            });
            this.$refs.formRef && (this.$refs.formRef.dispatchEvent(m), m.defaultPrevented) || (this.responseData = c, this.formState = "error");
          }
        );
      }
    }
  }
}), Vo = ["validate", "action", "method", "name"], Wo = {
  key: 3,
  class: "builder-form-error-text pre-3i4ezorcfhi"
};
function No(e, t, n, o, r, i) {
  var a;
  const s = g("Block"), l = g("Blocks");
  return d(), p("form", S({
    validate: e.validate,
    ref: "formRef",
    action: !e.sendWithJs && e.action,
    method: e.method,
    name: e.name,
    onSubmit: t[0] || (t[0] = (u) => e.onSubmit(u))
  }, { ...e.filterAttrs(e.attributes, "v-on:", !1) }, P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), [
    e.builderBlock && e.builderBlock.children ? (d(!0), p(I, { key: 0 }, N((a = e.builderBlock) == null ? void 0 : a.children, (u, h) => (d(), k(s, {
      key: `form-block-${e.idx}`,
      block: u,
      context: e.builderContext,
      registeredComponents: e.builderComponents,
      linkComponent: e.builderLinkComponent
    }, null, 8, ["block", "context", "registeredComponents", "linkComponent"]))), 128)) : v("", !0),
    e.submissionState() === "error" ? (d(), k(l, {
      key: 1,
      path: "errorMessage",
      blocks: e.errorMessage,
      context: e.builderContext
    }, null, 8, ["blocks", "context"])) : v("", !0),
    e.submissionState() === "sending" ? (d(), k(l, {
      key: 2,
      path: "sendingMessage",
      blocks: e.sendingMessage,
      context: e.builderContext
    }, null, 8, ["blocks", "context"])) : v("", !0),
    e.submissionState() === "error" && e.responseData ? (d(), p("pre", Wo, ne(JSON.stringify(e.responseData, null, 2)), 1)) : v("", !0),
    e.submissionState() === "success" ? (d(), k(l, {
      key: 4,
      path: "successMessage",
      blocks: e.successMessage,
      context: e.builderContext
    }, null, 8, ["blocks", "context"])) : v("", !0)
  ], 16, Vo);
}
const Fo = /* @__PURE__ */ b(Ao, [["render", No], ["__scopeId", "data-v-489330f7"]]), Lo = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Form:Input",
  image: "https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2Fad6f37889d9e40bbbbc72cdb5875d6ca",
  inputs: [
    {
      name: "type",
      type: "text",
      enum: ["text", "number", "email", "url", "checkbox", "radio", "range", "date", "datetime-local", "search", "tel", "time", "file", "month", "week", "password", "color", "hidden"],
      defaultValue: "text"
    },
    {
      name: "name",
      type: "string",
      required: !0,
      helperText: 'Every input in a form needs a unique name describing what it takes, e.g. "email"'
    },
    {
      name: "placeholder",
      type: "string",
      defaultValue: "Hello there",
      helperText: "Text to display when there is no value"
    },
    // TODO: handle value vs default value automatically like ng-model
    {
      name: "defaultValue",
      type: "string"
    },
    {
      name: "value",
      type: "string",
      advanced: !0
    },
    {
      name: "required",
      type: "boolean",
      helperText: "Is this input required to be filled out to submit a form",
      defaultValue: !1
    }
  ],
  noWrap: !0,
  static: !0,
  defaultStyles: {
    paddingTop: "10px",
    paddingBottom: "10px",
    paddingLeft: "10px",
    paddingRight: "10px",
    borderRadius: "3px",
    borderWidth: "1px",
    borderStyle: "solid",
    borderColor: "#ccc"
  }
}, Oo = f({
  name: "builder-form-input-component",
  props: [
    "attributes",
    "defaultValue",
    "placeholder",
    "type",
    "name",
    "value",
    "required"
  ],
  data() {
    return { isEditing: R, filterAttrs: x };
  }
}), Do = ["placeholder", "type", "name", "value", "defaultValue", "required"];
function Uo(e, t, n, o, r, i) {
  return d(), p("input", S({
    key: e.isEditing() && e.defaultValue ? e.defaultValue : "default-key",
    placeholder: e.placeholder,
    type: e.type,
    name: e.name,
    value: e.value,
    defaultValue: e.defaultValue,
    required: e.required
  }, e.filterAttrs(e.attributes, "v-on:", !1), P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), null, 16, Do);
}
const Mo = /* @__PURE__ */ b(Oo, [["render", Uo]]), jo = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Form:Select",
  image: "https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2F83acca093fb24aaf94dee136e9a4b045",
  defaultStyles: {
    alignSelf: "flex-start"
  },
  inputs: [{
    name: "options",
    type: "list",
    required: !0,
    subFields: [{
      name: "value",
      type: "text",
      required: !0
    }, {
      name: "name",
      type: "text"
    }],
    defaultValue: [{
      value: "option 1"
    }, {
      value: "option 2"
    }]
  }, {
    name: "name",
    type: "string",
    required: !0,
    helperText: 'Every select in a form needs a unique name describing what it gets, e.g. "email"'
  }, {
    name: "defaultValue",
    type: "string"
  }, {
    name: "value",
    type: "string",
    advanced: !0
  }, {
    name: "required",
    type: "boolean",
    defaultValue: !1
  }],
  static: !0,
  noWrap: !0
}, Ho = f({
  name: "builder-select-component",
  props: ["attributes", "value", "defaultValue", "name", "required", "options"],
  data() {
    return { isEditing: R, filterAttrs: x };
  }
}), Ko = ["value", "defaultValue", "name", "required"], qo = ["value"];
function zo(e, t, n, o, r, i) {
  return d(), p("select", S({
    value: e.value,
    key: e.isEditing() && e.defaultValue ? e.defaultValue : "default-key",
    defaultValue: e.defaultValue,
    name: e.name,
    required: e.required
  }, e.filterAttrs(e.attributes, "v-on:", !1), P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), [
    (d(!0), p(I, null, N(e.options, (s, l) => (d(), p("option", {
      key: `${s.name}-${l}`,
      value: s.value
    }, ne(s.name || s.value), 9, qo))), 128))
  ], 16, Ko);
}
const Jo = /* @__PURE__ */ b(Ho, [["render", zo]]), Go = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Form:SubmitButton",
  image: "https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2Fdf2820ffed1f4349a94c40b3221f5b98",
  defaultStyles: {
    appearance: "none",
    paddingTop: "15px",
    paddingBottom: "15px",
    paddingLeft: "25px",
    paddingRight: "25px",
    backgroundColor: "#3898EC",
    color: "white",
    borderRadius: "4px",
    cursor: "pointer"
  },
  inputs: [{
    name: "text",
    type: "text",
    defaultValue: "Click me"
  }],
  static: !0,
  noWrap: !0
  // TODO: optional children? maybe as optional form input
  // that only shows if advanced setting is flipped
  // TODO: defaultChildren
  // canHaveChildren: true,
}, Yo = f({
  name: "builder-submit-button",
  props: ["attributes", "text"],
  data() {
    return { filterAttrs: x };
  }
});
function Qo(e, t, n, o, r, i) {
  return d(), p("button", S({ type: "submit" }, e.filterAttrs(e.attributes, "v-on:", !1), P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), ne(e.text), 17);
}
const Xo = /* @__PURE__ */ b(Yo, [["render", Qo]]), Zo = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  name: "Form:TextArea",
  image: "https://cdn.builder.io/api/v1/image/assets%2FIsxPKMo2gPRRKeakUztj1D6uqed2%2Ff74a2f3de58c4c3e939204e5b6b8f6c3",
  inputs: [{
    advanced: !0,
    name: "value",
    type: "string"
  }, {
    name: "name",
    type: "string",
    required: !0,
    helperText: 'Every input in a form needs a unique name describing what it gets, e.g. "email"'
  }, {
    name: "defaultValue",
    type: "string"
  }, {
    name: "placeholder",
    type: "string",
    defaultValue: "Hello there"
  }, {
    name: "required",
    type: "boolean",
    defaultValue: !1
  }],
  defaultStyles: {
    paddingTop: "10px",
    paddingBottom: "10px",
    paddingLeft: "10px",
    paddingRight: "10px",
    borderRadius: "3px",
    borderWidth: "1px",
    borderStyle: "solid",
    borderColor: "#ccc"
  },
  static: !0,
  noWrap: !0
}, _o = f({
  name: "builder-textarea",
  props: [
    "attributes",
    "placeholder",
    "name",
    "value",
    "defaultValue",
    "required"
  ],
  data() {
    return { filterAttrs: x };
  }
}), er = ["placeholder", "name", "value", "defaultValue", "required"];
function tr(e, t, n, o, r, i) {
  return d(), p("textarea", S({
    placeholder: e.placeholder,
    name: e.name,
    value: e.value,
    defaultValue: e.defaultValue,
    required: e.required
  }, e.filterAttrs(e.attributes, "v-on:", !1), P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), null, 16, er);
}
const nr = /* @__PURE__ */ b(_o, [["render", tr]]), or = {
  shouldReceiveBuilderProps: {
    builderBlock: !1,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  },
  // friendlyName?
  name: "Raw:Img",
  hideFromInsertMenu: !0,
  image: "https://firebasestorage.googleapis.com/v0/b/builder-3b0a2.appspot.com/o/images%2Fbaseline-insert_photo-24px.svg?alt=media&token=4e5d0ef4-f5e8-4e57-b3a9-38d63a9b9dc4",
  inputs: [{
    name: "image",
    bubble: !0,
    type: "file",
    allowedFileTypes: ["jpeg", "jpg", "png", "svg", "gif", "webp"],
    required: !0
  }],
  noWrap: !0,
  static: !0
}, rr = f({
  name: "builder-img-component",
  props: [
    "backgroundSize",
    "backgroundPosition",
    "imgSrc",
    "altText",
    "image",
    "attributes"
  ],
  data() {
    return { isEditing: R, filterAttrs: x };
  }
}), ir = ["alt", "src"];
function sr(e, t, n, o, r, i) {
  return d(), p("img", S({
    style: {
      objectFit: e.backgroundSize || "cover",
      objectPosition: e.backgroundPosition || "center"
    },
    key: e.isEditing() && e.imgSrc || "default-key",
    alt: e.altText,
    src: e.imgSrc || e.image
  }, e.filterAttrs(e.attributes, "v-on:", !1), P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), null, 16, ir);
}
const ar = /* @__PURE__ */ b(rr, [["render", sr]]), lr = {
  name: "Video",
  canHaveChildren: !0,
  defaultStyles: {
    minHeight: "20px",
    minWidth: "20px"
  },
  image: "https://firebasestorage.googleapis.com/v0/b/builder-3b0a2.appspot.com/o/images%2Fbaseline-videocam-24px%20(1).svg?alt=media&token=49a84e4a-b20e-4977-a650-047f986874bb",
  inputs: [{
    name: "video",
    type: "file",
    allowedFileTypes: ["mp4"],
    bubble: !0,
    defaultValue: "https://cdn.builder.io/o/assets%2FYJIGb4i01jvw0SRdL5Bt%2Fd27731a526464deba0016216f5f9e570%2Fcompressed?apiKey=YJIGb4i01jvw0SRdL5Bt&token=d27731a526464deba0016216f5f9e570&alt=media&optimized=true",
    required: !0
  }, {
    name: "posterImage",
    type: "file",
    allowedFileTypes: ["jpeg", "png"],
    helperText: "Image to show before the video plays"
  }, {
    name: "autoPlay",
    type: "boolean",
    defaultValue: !0
  }, {
    name: "controls",
    type: "boolean",
    defaultValue: !1
  }, {
    name: "muted",
    type: "boolean",
    defaultValue: !0
  }, {
    name: "loop",
    type: "boolean",
    defaultValue: !0
  }, {
    name: "playsInline",
    type: "boolean",
    defaultValue: !0
  }, {
    name: "fit",
    type: "text",
    defaultValue: "cover",
    enum: ["contain", "cover", "fill", "auto"]
  }, {
    name: "preload",
    type: "text",
    defaultValue: "metadata",
    enum: ["auto", "metadata", "none"]
  }, {
    name: "fitContent",
    type: "boolean",
    helperText: "When child blocks are provided, fit to them instead of using the aspect ratio",
    defaultValue: !0,
    advanced: !0
  }, {
    name: "position",
    type: "text",
    defaultValue: "center",
    enum: ["center", "top", "left", "right", "bottom", "top left", "top right", "bottom left", "bottom right"]
  }, {
    name: "height",
    type: "number",
    advanced: !0
  }, {
    name: "width",
    type: "number",
    advanced: !0
  }, {
    name: "aspectRatio",
    type: "number",
    advanced: !0,
    defaultValue: 0.7004048582995948
  }, {
    name: "lazyLoad",
    type: "boolean",
    helperText: 'Load this video "lazily" - as in only when a user scrolls near the video. Recommended for optmized performance and bandwidth consumption',
    defaultValue: !0,
    advanced: !0
  }],
  shouldReceiveBuilderProps: {
    builderBlock: !0,
    builderContext: !1,
    builderComponents: !1,
    builderLinkComponent: !1
  }
}, cr = f({
  name: "builder-video",
  props: [
    "autoPlay",
    "muted",
    "controls",
    "loop",
    "playsInline",
    "preload",
    "attributes",
    "fit",
    "position",
    "aspectRatio",
    "video",
    "posterImage",
    "lazyLoad",
    "fitContent",
    "builderBlock"
  ],
  computed: {
    videoProps() {
      return {
        ...this.autoPlay === !0 ? {
          autoPlay: !0
        } : {},
        ...this.muted === !0 ? {
          muted: !0
        } : {},
        ...this.controls === !0 ? {
          controls: !0
        } : {},
        ...this.loop === !0 ? {
          loop: !0
        } : {},
        ...this.playsInline === !0 ? {
          playsInline: !0
        } : {}
      };
    },
    spreadProps() {
      return {
        ...this.videoProps
      };
    }
  }
}), dr = { style: {
  position: "relative"
} }, ur = ["preload", "src", "poster"], pr = ["src"], mr = {
  key: 1,
  style: {
    display: "flex",
    flexDirection: "column",
    alignItems: "stretch"
  }
}, hr = {
  key: 2,
  style: {
    pointerEvents: "none",
    display: "flex",
    flexDirection: "column",
    alignItems: "stretch",
    position: "absolute",
    top: "0",
    left: "0",
    width: "100%",
    height: "100%"
  }
};
function fr(e, t, n, o, r, i) {
  var s, l, a, u, h, c, m;
  return d(), p("div", dr, [
    z("video", S({
      class: "builder-video",
      preload: e.preload || "metadata",
      style: {
        width: "100%",
        height: "100%",
        ...(s = e.attributes) == null ? void 0 : s.style,
        objectFit: e.fit,
        objectPosition: e.position,
        // Hack to get object fit to work as expected and
        // not have the video overflow
        zIndex: 2,
        borderRadius: "1px",
        ...e.aspectRatio ? {
          position: "absolute"
        } : null
      },
      src: e.video || "no-src",
      poster: e.posterImage
    }, e.spreadProps), [
      e.lazyLoad ? v("", !0) : (d(), p("source", {
        key: 0,
        type: "video/mp4",
        src: e.video
      }, null, 8, pr))
    ], 16, ur),
    e.aspectRatio && !(e.fitContent && ((a = (l = e.builderBlock) == null ? void 0 : l.children) != null && a.length)) ? (d(), p("div", {
      key: 0,
      style: O({
        width: "100%",
        paddingTop: e.aspectRatio * 100 + "%",
        pointerEvents: "none",
        fontSize: "0px"
      })
    }, null, 4)) : v("", !0),
    (h = (u = e.builderBlock) == null ? void 0 : u.children) != null && h.length && e.fitContent ? (d(), p("div", mr, [
      E(e.$slots, "default")
    ])) : v("", !0),
    (m = (c = e.builderBlock) == null ? void 0 : c.children) != null && m.length && !e.fitContent ? (d(), p("div", hr, [
      E(e.$slots, "default")
    ])) : v("", !0)
  ]);
}
const br = /* @__PURE__ */ b(cr, [["render", fr]]), gr = () => [{
  component: vo,
  ...bo
}, {
  component: Ro,
  ...Co
}, {
  component: Fo,
  ...Bo
}, {
  component: Mo,
  ...Lo
}, {
  component: Xo,
  ...Go
}, {
  component: Jo,
  ...jo
}, {
  component: nr,
  ...Zo
}, {
  component: ar,
  ...or
}, {
  component: br,
  ...lr
}], Te = () => [{
  component: wt,
  ...Gn
}, {
  component: bn,
  ...Yn
}, {
  component: kn,
  ...Qn
}, {
  component: Rn,
  ...Xn
}, {
  component: En,
  ...Zn
}, {
  component: no,
  ..._n
}, {
  component: $i,
  ...oo
}, {
  component: fo,
  ...uo
}, {
  component: co,
  ...ro
}, {
  component: Jn,
  ...jn
}, ...gr()], yr = (e) => ({
  type: "builder.registerComponent",
  data: ue(e)
}), kr = (e) => {
  const t = e.toString().trim();
  return `return (${!t.startsWith("function") && !t.startsWith("(") ? "function " : ""}${t}).apply(this, arguments)`;
}, vr = (e) => typeof e == "function" ? kr(e) : Y(e), ue = ({
  inputs: e,
  ...t
}) => ({
  ...Y(t),
  inputs: e == null ? void 0 : e.map((n) => Object.entries(n).reduce((o, [r, i]) => ({
    ...o,
    [r]: vr(i)
  }), {}))
}), Cr = `function updateCookiesAndStyles(contentId, variants, isHydrationTarget) {
  function getAndSetVariantId() {
    function setCookie(name, value, days) {
      let expires = '';
      if (days) {
        const date = new Date();
        date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000);
        expires = '; expires=' + date.toUTCString();
      }
      document.cookie = name + '=' + (value || '') + expires + '; path=/' + '; Secure; SameSite=None';
    }
    function getCookie(name) {
      const nameEQ = name + '=';
      const ca = document.cookie.split(';');
      for (let i = 0; i < ca.length; i++) {
        let c = ca[i];
        while (c.charAt(0) === ' ') c = c.substring(1, c.length);
        if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length, c.length);
      }
      return null;
    }
    const cookieName = \`builder.tests.\${contentId}\`;
    const variantInCookie = getCookie(cookieName);
    const availableIDs = variants.map(vr => vr.id).concat(contentId);
    if (variantInCookie && availableIDs.includes(variantInCookie)) {
      return variantInCookie;
    }
    let n = 0;
    const random = Math.random();
    for (let i = 0; i < variants.length; i++) {
      const variant = variants[i];
      const testRatio = variant.testRatio;
      n += testRatio;
      if (random < n) {
        setCookie(cookieName, variant.id);
        return variant.id;
      }
    }
    setCookie(cookieName, contentId);
    return contentId;
  }
  const winningVariantId = getAndSetVariantId();
  const styleEl = document.currentScript?.previousElementSibling;
  if (isHydrationTarget) {
    styleEl.remove();
    const thisScriptEl = document.currentScript;
    thisScriptEl?.remove();
  } else {
    const newStyleStr = variants.concat({
      id: contentId
    }).filter(variant => variant.id !== winningVariantId).map(value => {
      return \`.variant-\${value.id} {  display: none; }
        \`;
    }).join('');
    styleEl.innerHTML = newStyleStr;
  }
}`, Sr = `function updateVariantVisibility(variantContentId, defaultContentId, isHydrationTarget) {
  if (!navigator.cookieEnabled) {
    return;
  }
  function getCookie(name) {
    const nameEQ = name + '=';
    const ca = document.cookie.split(';');
    for (let i = 0; i < ca.length; i++) {
      let c = ca[i];
      while (c.charAt(0) === ' ') c = c.substring(1, c.length);
      if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length, c.length);
    }
    return null;
  }
  const cookieName = \`builder.tests.\${defaultContentId}\`;
  const winningVariant = getCookie(cookieName);
  const parentDiv = document.currentScript?.parentElement;
  const isDefaultContent = variantContentId === defaultContentId;
  const isWinningVariant = winningVariant === variantContentId;
  if (isWinningVariant && !isDefaultContent) {
    parentDiv?.removeAttribute('hidden');
    parentDiv?.removeAttribute('aria-hidden');
  } else if (!isWinningVariant && isDefaultContent) {
    parentDiv?.setAttribute('hidden', 'true');
    parentDiv?.setAttribute('aria-hidden', 'true');
  }
  if (isHydrationTarget) {
    if (!isWinningVariant) {
      parentDiv?.remove();
    }
    const thisScriptEl = document.currentScript;
    thisScriptEl?.remove();
  }
  return;
}`, Ze = "builderIoAbTest", _e = "builderIoRenderContent", _ = (e) => Object.values((e == null ? void 0 : e.variations) || {}).map((t) => ({
  ...t,
  testVariationId: t.id,
  id: e == null ? void 0 : e.id
})), wr = ({
  canTrack: e,
  content: t
}) => !(!(_(t).length > 0) || !e), Tr = (e) => e === "react" || e === "reactNative", et = Tr(A), Ir = () => `
  window.${Ze} = ${Cr}
  window.${_e} = ${Sr}
  `, $r = (e, t) => `
  window.${Ze}(
    "${t}",${JSON.stringify(e)}, ${et}
  )`, Rr = ({
  contentId: e,
  variationId: t
}) => `window.${_e}(
    "${t}", "${e}", ${et}
  )`, tt = f({
  render() {
    return Ae("script", {
      innerHTML: this.scriptStr,
      "data-id": this.id
    });
  },
  name: "inlined-script",
  props: ["scriptStr", "id"]
});
function Br() {
  return typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : globalThis;
}
function Pr() {
  const e = Br().fetch;
  if (typeof e > "u")
    throw console.warn(`Builder SDK could not find a global fetch function. Make sure you have a polyfill for fetch in your project. 
      For more information, read https://github.com/BuilderIO/this-package-uses-fetch`), new Error("Builder SDK could not find a global `fetch` function");
  return e;
}
const nt = Pr();
function pe(e, t = null, n = ".") {
  return Object.keys(e).reduce((o, r) => {
    const i = e[r], s = [t, r].filter(Boolean).join(n);
    return [typeof i == "object", i !== null, !(Array.isArray(i) && i.length === 0)].every(Boolean) ? {
      ...o,
      ...pe(i, s, n)
    } : {
      ...o,
      [s]: i
    };
  }, {});
}
function ot(e, t, n = {}) {
  for (const o in e) {
    const r = e[o], i = t ? t + "." + o : o;
    r && typeof r == "object" && !Array.isArray(r) && !Object.keys(r).find((s) => s.startsWith("$")) ? ot(r, i, n) : n[i] = r;
  }
  return n;
}
const Er = "v3", Ie = "builder.", xr = "options.", rt = (e) => {
  if (!e)
    return {};
  const t = We(e), n = {};
  return Object.keys(t).forEach((o) => {
    if (o.startsWith(Ie)) {
      const r = o.replace(Ie, "").replace(xr, "");
      n[r] = t[o];
    }
  }), n;
}, Ar = () => {
  if (!T())
    return {};
  const e = new URLSearchParams(window.location.search);
  return rt(e);
}, $e = (e) => typeof e == "number" && !isNaN(e) && e >= 0, ke = (e) => {
  const {
    limit: t = 30,
    userAttributes: n,
    query: o,
    model: r,
    apiKey: i,
    enrich: s,
    locale: l,
    apiVersion: a = Er,
    fields: u,
    omit: h,
    offset: c,
    cacheSeconds: m,
    staleCacheSeconds: C,
    sort: y,
    includeUnpublished: F
  } = e;
  if (!i)
    throw new Error("Missing API key");
  if (!["v3"].includes(a))
    throw new Error(`Invalid apiVersion: expected 'v3', received '${a}'`);
  const X = t !== 1, w = new URL(`https://cdn.builder.io/api/${a}/content/${r}`);
  if (w.searchParams.set("apiKey", i), w.searchParams.set("limit", String(t)), w.searchParams.set("noTraverse", String(X)), w.searchParams.set("includeRefs", String(!0)), l && w.searchParams.set("locale", l), s && w.searchParams.set("enrich", String(s)), w.searchParams.set("omit", h || "meta.componentsUsed"), u && w.searchParams.set("fields", u), Number.isFinite(c) && c > -1 && w.searchParams.set("offset", String(Math.floor(c))), typeof F == "boolean" && w.searchParams.set("includeUnpublished", String(F)), m && $e(m) && w.searchParams.set("cacheSeconds", String(m)), C && $e(C) && w.searchParams.set("staleCacheSeconds", String(C)), y) {
    const L = pe({
      sort: y
    });
    for (const j in L)
      w.searchParams.set(j, JSON.stringify(L[j]));
  }
  const pt = {
    ...Ar(),
    ...We(e.options || {})
  }, Ce = pe(pt);
  for (const L in Ce)
    w.searchParams.set(L, String(Ce[L]));
  if (n && w.searchParams.set("userAttributes", JSON.stringify(n)), o) {
    const L = ot({
      query: o
    });
    for (const j in L)
      w.searchParams.set(j, JSON.stringify(L[j]));
  }
  return w;
}, Vr = (e) => "results" in e;
async function ve(e) {
  const t = await Fr({
    ...e,
    limit: 1
  });
  return t && t[0] || null;
}
const Wr = async (e) => {
  const t = ke(e);
  return await (await (e.fetch ?? nt)(t.href, e.fetchOptions)).json();
}, Nr = async (e, t, n = ke(e)) => {
  const o = G(e.canTrack);
  if (n.search.includes("preview="), !o || !(T() || A === "reactNative"))
    return t.results;
  try {
    const r = [];
    for (const i of t.results)
      r.push(await Mn({
        item: i,
        canTrack: o
      }));
    t.results = r;
  } catch (r) {
    B.error("Could not process A/B tests. ", r);
  }
  return t.results;
};
async function Fr(e) {
  try {
    const t = ke(e), n = await Wr(e);
    return Vr(n) ? Nr(e, n) : (B.error("Error fetching data. ", {
      url: t,
      content: n,
      options: e
    }), null);
  } catch (t) {
    return B.error("Error fetching data. ", t), null;
  }
}
function Lr(e) {
  const t = e || (T() ? window.location.search : void 0);
  return t ? Ne(t).indexOf("builder.preview=") !== -1 : !1;
}
function Or() {
  return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function(e) {
    const t = Math.random() * 16 | 0;
    return (e == "x" ? t : t & 3 | 8).toString(16);
  });
}
function it() {
  return Or().replace(/-/g, "");
}
const st = "builderSessionId", Dr = async ({
  canTrack: e
}) => {
  if (!e)
    return;
  const t = await Je({
    name: st,
    canTrack: e
  });
  if (D(t))
    return t;
  {
    const n = Ur();
    return Mr({
      id: n,
      canTrack: e
    }), n;
  }
}, Ur = () => it(), Mr = ({
  id: e,
  canTrack: t
}) => Ge({
  name: st,
  value: e,
  canTrack: t
}), at = () => T() && typeof localStorage < "u" ? localStorage : void 0, jr = ({
  key: e,
  canTrack: t
}) => {
  var n;
  try {
    return t ? (n = at()) == null ? void 0 : n.getItem(e) : void 0;
  } catch (o) {
    console.debug("[LocalStorage] GET error: ", o);
    return;
  }
}, Hr = ({
  key: e,
  canTrack: t,
  value: n
}) => {
  var o;
  try {
    t && ((o = at()) == null || o.setItem(e, n));
  } catch (r) {
    console.debug("[LocalStorage] SET error: ", r);
  }
}, lt = "builderVisitorId", Kr = ({
  canTrack: e
}) => {
  if (!e)
    return;
  const t = jr({
    key: lt,
    canTrack: e
  });
  if (D(t))
    return t;
  {
    const n = qr();
    return zr({
      id: n,
      canTrack: e
    }), n;
  }
}, qr = () => it(), zr = ({
  id: e,
  canTrack: t
}) => Hr({
  key: lt,
  value: e,
  canTrack: t
}), Jr = async ({
  canTrack: e
}) => {
  if (!e)
    return {
      visitorId: void 0,
      sessionId: void 0
    };
  const t = await Dr({
    canTrack: e
  }), n = Kr({
    canTrack: e
  });
  return {
    sessionId: t,
    visitorId: n
  };
}, Gr = async ({
  type: e,
  canTrack: t,
  apiKey: n,
  metadata: o,
  ...r
}) => ({
  type: e,
  data: {
    ...r,
    metadata: {
      url: location.href,
      ...o
    },
    ...await Jr({
      canTrack: t
    }),
    userAttributes: Fe(),
    ownerId: n
  }
});
async function me(e) {
  if (!e.apiKey) {
    B.error("Missing API key for track call. Please provide your API key.");
    return;
  }
  if (e.canTrack && !R() && (T() || A === "reactNative"))
    return fetch("https://cdn.builder.io/api/v1/track", {
      method: "POST",
      body: JSON.stringify({
        events: [await Gr(e)]
      }),
      headers: {
        "content-type": "application/json"
      },
      mode: "cors"
    }).catch((t) => {
      console.error("Failed to track: ", t);
    });
}
const Ni = (e) => me({
  ...e,
  canTrack: !0
});
function Re(e) {
  return Math.round(e * 1e3) / 1e3;
}
const Yr = (e, t, n = !0) => {
  if (!(e instanceof HTMLElement))
    return null;
  let o = n ? e : e.parentElement;
  do {
    if (!o)
      return null;
    if (t(o))
      return o;
  } while (o = o.parentElement);
  return null;
}, Qr = (e) => Yr(e, (t) => {
  const n = t.getAttribute("builder-id") || t.id;
  return (n == null ? void 0 : n.indexOf("builder-")) === 0;
}), Be = ({
  event: e,
  target: t
}) => {
  const n = t.getBoundingClientRect(), o = e.clientX - n.left, r = e.clientY - n.top, i = Re(o / n.width), s = Re(r / n.height);
  return {
    x: i,
    y: s
  };
}, Xr = (e) => {
  const t = e.target, n = t && Qr(t), o = (n == null ? void 0 : n.getAttribute("builder-id")) || (n == null ? void 0 : n.id);
  return {
    targetBuilderElement: o || void 0,
    metadata: {
      targetOffset: t ? Be({
        event: e,
        target: t
      }) : void 0,
      builderTargetOffset: n ? Be({
        event: e,
        target: n
      }) : void 0,
      builderElementIndex: n && o ? [].slice.call(document.getElementsByClassName(o)).indexOf(n) : void 0
    }
  };
}, Zr = ["*.beta.builder.io", "beta.builder.io", "builder.io", "localhost", "qa.builder.io"];
function ct(e, t) {
  if (!t.origin.startsWith("http") && !t.origin.startsWith("https"))
    return !1;
  const n = new URL(t.origin), o = n.hostname;
  return (e || Zr).findIndex((r) => r.startsWith("*.") ? o.endsWith(r.slice(1)) : r === o) > -1;
}
const _r = "1.1.2", Pe = {};
function ei(e, t) {
  let n = Pe[e];
  if (n || (n = Pe[e] = []), n.push(t), T()) {
    const o = {
      type: "builder.register",
      data: {
        type: e,
        info: t
      }
    };
    try {
      parent.postMessage(o, "*"), parent !== window && window.postMessage(o, "*");
    } catch (r) {
      console.debug("Could not postmessage", r);
    }
  }
}
const ti = () => {
  ei("insertMenu", {
    name: "_default",
    default: !0,
    items: [{
      name: "Box"
    }, {
      name: "Text"
    }, {
      name: "Image"
    }, {
      name: "Columns"
    }, {
      name: "Core:Section"
    }, {
      name: "Core:Button"
    }, {
      name: "Embed"
    }, {
      name: "Custom Code"
    }]
  });
};
let Ee = !1;
const dt = (e = {}) => {
  var t, n;
  Ee || (Ee = !0, T() && ((t = window.parent) == null || t.postMessage({
    type: "builder.sdkInfo",
    data: {
      target: A,
      version: _r,
      supportsPatchUpdates: !1,
      // Supports builder-model="..." attribute which is needed to
      // scope our '+ add block' button styling
      supportsAddBlockScoping: !0,
      supportsCustomBreakpoints: !0
    }
  }, "*"), (n = window.parent) == null || n.postMessage({
    type: "builder.updateContent",
    data: {
      options: e
    }
  }, "*"), window.addEventListener("message", (o) => {
    var i, s;
    if (!ct(e.trustedHosts, o))
      return;
    const {
      data: r
    } = o;
    if (r != null && r.type)
      switch (r.type) {
        case "builder.evaluate": {
          const l = r.data.text, a = r.data.arguments || [], u = r.data.id, h = new Function(l);
          let c, m = null;
          try {
            c = h.apply(null, a);
          } catch (C) {
            m = C;
          }
          m ? (i = window.parent) == null || i.postMessage({
            type: "builder.evaluateError",
            data: {
              id: u,
              error: m.message
            }
          }, "*") : c && typeof c.then == "function" ? c.then((C) => {
            var y;
            (y = window.parent) == null || y.postMessage({
              type: "builder.evaluateResult",
              data: {
                id: u,
                result: C
              }
            }, "*");
          }).catch(console.error) : (s = window.parent) == null || s.postMessage({
            type: "builder.evaluateResult",
            data: {
              result: c,
              id: u
            }
          }, "*");
          break;
        }
      }
  })));
}, ut = ({
  model: e,
  trustedHosts: t,
  callbacks: n
}) => (o) => {
  if (!ct(t, o))
    return;
  const {
    data: r
  } = o;
  if (r)
    switch (r.type) {
      case "builder.configureSdk": {
        n.configureSdk(r.data);
        break;
      }
      case "builder.triggerAnimation": {
        n.animation(r.data);
        break;
      }
      case "builder.contentUpdate": {
        const i = r.data, s = i.key || i.alias || i.entry || i.modelName, l = i.data;
        s === e && n.contentUpdate(l);
        break;
      }
    }
}, Fi = (e, t, n) => {
  if (!T)
    return B.warn("`subscribeToEditor` only works in the browser. It currently seems to be running on the server."), () => {
    };
  dt();
  const o = ut({
    callbacks: {
      contentUpdate: t,
      animation: () => {
      },
      configureSdk: () => {
      }
    },
    model: e,
    trustedHosts: n == null ? void 0 : n.trustedHosts
  });
  return window.addEventListener("message", o), () => {
    window.removeEventListener("message", o);
  };
}, ni = (e) => {
  var i;
  const t = e.family + (e.kind && !e.kind.includes("#") ? ", " + e.kind : ""), n = t.split(",")[0], o = e.fileUrl ?? ((i = e == null ? void 0 : e.files) == null ? void 0 : i.regular);
  let r = "";
  if (o && t && n && (r += `
@font-face {
font-family: "${t}";
src: local("${n}"), url('${o}') format('woff2');
font-display: fallback;
font-weight: 400;
}
      `.trim()), e.files)
    for (const s in e.files) {
      if (!(String(Number(s)) === s))
        continue;
      const a = e.files[s];
      a && a !== o && (r += `
@font-face {
font-family: "${t}";
src: url('${a}') format('woff2');
font-display: fallback;
font-weight: ${s};
}
        `.trim());
    }
  return r;
}, oi = ({
  customFonts: e
}) => {
  var t;
  return ((t = e == null ? void 0 : e.map((n) => ni(n))) == null ? void 0 : t.join(" ")) || "";
}, ri = ({
  cssCode: e,
  contentId: t
}) => e ? t ? (e == null ? void 0 : e.replace(/&/g, `div[builder-content-id="${t}"]`)) || "" : e : "", ii = `
.builder-button {
  all: unset;
}

.builder-text > p:first-of-type, .builder-text > .builder-paragraph:first-of-type {
  margin: 0;
}
.builder-text > p, .builder-text > .builder-paragraph {
  color: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  font-weight: inherit;
  font-size: inherit;
  text-align: inherit;
  font-family: inherit;
}
`, si = (e) => e ? "" : ii, ai = (e) => `variant-${e}`, li = f({
  name: "enable-editor",
  components: { DynamicDiv: Q },
  props: [
    "builderContextSignal",
    "canTrack",
    "apiKey",
    "locale",
    "enrich",
    "trustedHosts",
    "model",
    "content",
    "data",
    "showContent",
    "contentWrapper",
    "context",
    "contentWrapperProps"
  ],
  data() {
    return {
      ContentWrapper: this.contentWrapper || "div",
      httpReqsData: {},
      httpReqsPending: {},
      clicked: !1,
      builderContext: de,
      getWrapperClassName: ai
    };
  },
  provide() {
    const e = this;
    return {
      [de.key]: e.builderContextSignal
    };
  },
  mounted() {
    (() => {
      var n, o;
      if (T()) {
        if (R() && this.$refs.elementRef && this.$refs.elementRef.dispatchEvent(
          new CustomEvent("initeditingbldr")
        ), this.builderContextSignal.content && G(this.canTrack)) {
          const i = (n = this.builderContextSignal.content) == null ? void 0 : n.testVariationId, s = (o = this.builderContextSignal.content) == null ? void 0 : o.id, l = this.apiKey;
          me({
            type: "impression",
            canTrack: !0,
            contentId: s,
            apiKey: l,
            variationId: i !== s ? i : void 0
          });
        }
        Lr() && !R() && this.$refs.elementRef && this.$refs.elementRef.dispatchEvent(
          new CustomEvent("initpreviewingbldr")
        );
      }
    })(), (() => {
      this.apiKey || B.error(
        "No API key provided to `Content` component. This can cause issues. Please provide an API key using the `apiKey` prop."
      ), this.evaluateJsCode(), this.runHttpRequests(), this.emitStateUpdate();
    })();
  },
  watch: {
    onUpdateHook0: {
      handler() {
        this.content && this.mergeNewContent(this.content);
      },
      immediate: !0
    },
    onUpdateHook1: {
      handler() {
        this.evaluateJsCode();
      },
      immediate: !0
    },
    onUpdateHook2: {
      handler() {
        this.runHttpRequests();
      },
      immediate: !0
    },
    onUpdateHook3: {
      handler() {
        this.emitStateUpdate();
      },
      immediate: !0
    },
    onUpdateHook4: {
      handler() {
        this.data && this.mergeNewRootState(this.data);
      },
      immediate: !0
    },
    onUpdateHook5: {
      handler() {
        this.locale && this.mergeNewRootState({
          locale: this.locale
        });
      },
      immediate: !0
    }
  },
  unmounted() {
    T() && (window.removeEventListener("message", this.processMessage), window.removeEventListener(
      "builder:component:stateChangeListenerActivated",
      this.emitStateUpdate
    ));
  },
  computed: {
    showContentProps() {
      return this.showContent ? {} : {
        hidden: !0,
        "aria-hidden": !0
      };
    },
    onUpdateHook0() {
      return {
        0: this.content
      };
    },
    onUpdateHook1() {
      var e, t;
      return {
        0: (t = (e = this.builderContextSignal.content) == null ? void 0 : e.data) == null ? void 0 : t.jsCode
      };
    },
    onUpdateHook2() {
      var e, t;
      return {
        0: (t = (e = this.builderContextSignal.content) == null ? void 0 : e.data) == null ? void 0 : t.httpRequests
      };
    },
    onUpdateHook3() {
      return {
        0: this.builderContextSignal.rootState
      };
    },
    onUpdateHook4() {
      return {
        0: this.data
      };
    },
    onUpdateHook5() {
      return {
        0: this.locale
      };
    }
  },
  methods: {
    mergeNewRootState(e) {
      var n, o;
      const t = {
        ...this.builderContextSignal.rootState,
        ...e
      };
      this.builderContextSignal.rootSetState ? (o = (n = this.builderContextSignal).rootSetState) == null || o.call(n, t) : this.builderContextSignal.rootState = t;
    },
    mergeNewContent(e) {
      var n, o, r, i, s;
      const t = {
        ...this.builderContextSignal.content,
        ...e,
        data: {
          ...(n = this.builderContextSignal.content) == null ? void 0 : n.data,
          ...e == null ? void 0 : e.data
        },
        meta: {
          ...(o = this.builderContextSignal.content) == null ? void 0 : o.meta,
          ...e == null ? void 0 : e.meta,
          breakpoints: ((r = e == null ? void 0 : e.meta) == null ? void 0 : r.breakpoints) || ((s = (i = this.builderContextSignal.content) == null ? void 0 : i.meta) == null ? void 0 : s.breakpoints)
        }
      };
      this.builderContextSignal.content = t;
    },
    processMessage(e) {
      return ut({
        model: this.model,
        trustedHosts: this.trustedHosts,
        callbacks: {
          configureSdk: (t) => {
            var r;
            const { breakpoints: n, contentId: o } = t;
            !o || o !== ((r = this.builderContextSignal.content) == null ? void 0 : r.id) || n && this.mergeNewContent({
              meta: {
                breakpoints: n
              }
            });
          },
          animation: (t) => {
            je(t);
          },
          contentUpdate: (t) => {
            this.mergeNewContent(t);
          }
        }
      })(e);
    },
    evaluateJsCode() {
      var t, n;
      const e = (n = (t = this.builderContextSignal.content) == null ? void 0 : t.data) == null ? void 0 : n.jsCode;
      e && te({
        code: e,
        context: this.context || {},
        localState: void 0,
        rootState: this.builderContextSignal.rootState,
        rootSetState: this.builderContextSignal.rootSetState,
        /**
         * We don't want to cache the result of the JS code, since it's arbitrary side effect code.
         */
        enableCache: !1
      });
    },
    onClick(e) {
      var t, n;
      if (this.builderContextSignal.content) {
        const o = (t = this.builderContextSignal.content) == null ? void 0 : t.testVariationId, r = (n = this.builderContextSignal.content) == null ? void 0 : n.id;
        me({
          type: "click",
          canTrack: G(this.canTrack),
          contentId: r,
          apiKey: this.apiKey,
          variationId: o !== r ? o : void 0,
          ...Xr(e),
          unique: !this.clicked
        });
      }
      this.clicked || (this.clicked = !0);
    },
    runHttpRequests() {
      var t, n;
      const e = ((n = (t = this.builderContextSignal.content) == null ? void 0 : t.data) == null ? void 0 : n.httpRequests) ?? {};
      Object.entries(e).forEach(([o, r]) => {
        if (!r || this.httpReqsPending[o] || this.httpReqsData[o] && !R())
          return;
        this.httpReqsPending[o] = !0;
        const i = r.replace(
          /{{([^}]+)}}/g,
          (s, l) => String(
            te({
              code: l,
              context: this.context || {},
              localState: void 0,
              rootState: this.builderContextSignal.rootState,
              rootSetState: this.builderContextSignal.rootSetState,
              enableCache: !0
            })
          )
        );
        nt(i).then((s) => s.json()).then((s) => {
          this.mergeNewRootState({
            [o]: s
          }), this.httpReqsData[o] = !0;
        }).catch((s) => {
          console.error("error fetching dynamic data", r, s);
        }).finally(() => {
          this.httpReqsPending[o] = !1;
        });
      });
    },
    emitStateUpdate() {
      R() && window.dispatchEvent(
        new CustomEvent(
          "builder:component:stateChange",
          {
            detail: {
              state: Y(this.builderContextSignal.rootState),
              ref: {
                name: this.model
              }
            }
          }
        )
      );
    },
    elementRef_onIniteditingbldr(e) {
      window.addEventListener("message", this.processMessage), ti(), dt({
        ...this.locale ? {
          locale: this.locale
        } : {},
        ...this.enrich ? {
          enrich: this.enrich
        } : {},
        ...this.trustedHosts ? {
          trustedHosts: this.trustedHosts
        } : {}
      }), Object.values(
        this.builderContextSignal.componentInfos
      ).forEach((t) => {
        var o;
        const n = yr(t);
        (o = window.parent) == null || o.postMessage(n, "*");
      }), window.addEventListener(
        "builder:component:stateChangeListenerActivated",
        this.emitStateUpdate
      );
    },
    elementRef_onInitpreviewingbldr(e) {
      const t = new URL(location.href).searchParams, n = t.get("builder.preview"), o = t.get(
        `builder.overrides.${n}`
      ), r = t.get("apiKey") || t.get("builder.space");
      n === this.model && r === this.apiKey && (!this.content || o === this.content.id) && ve({
        model: this.model,
        apiKey: this.apiKey,
        apiVersion: this.builderContextSignal.apiVersion
      }).then((i) => {
        i && this.mergeNewContent(i);
      });
    }
  }
});
function ci(e, t, n, o, r, i) {
  var s, l, a;
  return e.builderContextSignal.content ? (d(), k(K(e.ContentWrapper), S({
    key: 0,
    ref: "elementRef",
    onClick: (u) => e.onClick(u),
    "builder-content-id": (s = e.builderContextSignal.content) == null ? void 0 : s.id,
    "builder-model": e.model,
    class: e.getWrapperClassName(((l = e.content) == null ? void 0 : l.testVariationId) || ((a = e.content) == null ? void 0 : a.id)),
    onIniteditingbldr: (u) => e.elementRef_onIniteditingbldr(u),
    onInitpreviewingbldr: (u) => e.elementRef_onInitpreviewingbldr(u)
  }, { ...e.showContentProps, ...e.contentWrapperProps }), {
    default: W(() => [
      E(e.$slots, "default")
    ]),
    _: 3
  }, 16, ["onClick", "builder-content-id", "builder-model", "class", "onIniteditingbldr", "onInitpreviewingbldr"])) : v("", !0);
}
const di = /* @__PURE__ */ b(li, [["render", ci]]), ui = f({
  name: "content-styles",
  components: { InlinedStyles: ge },
  props: ["cssCode", "contentId", "customFonts", "isNestedRender"],
  data() {
    return {
      injectedStyles: `
${ri({
        cssCode: this.cssCode,
        contentId: this.contentId
      })}
${oi({
        customFonts: this.customFonts
      })}
${si(this.isNestedRender)}
`.trim()
    };
  }
});
function pi(e, t, n, o, r, i) {
  const s = g("InlinedStyles");
  return d(), k(s, {
    id: "builderio-content",
    styles: e.injectedStyles
  }, null, 8, ["styles"]);
}
const mi = /* @__PURE__ */ b(ui, [["render", pi]]), hi = ({
  content: e,
  data: t,
  locale: n
}) => {
  var i, s, l;
  const o = {}, r = ((i = e == null ? void 0 : e.data) == null ? void 0 : i.state) || {};
  return (l = (s = e == null ? void 0 : e.data) == null ? void 0 : s.inputs) == null || l.forEach((a) => {
    a.name && a.defaultValue !== void 0 && (o[a.name] = a.defaultValue);
  }), {
    ...o,
    ...r,
    ...t,
    ...n ? {
      locale: n
    } : {}
  };
}, fi = ({
  content: e,
  data: t
}) => e ? {
  ...e,
  data: {
    ...e == null ? void 0 : e.data,
    ...t
  },
  meta: e == null ? void 0 : e.meta
} : void 0, bi = bt, gi = f({
  name: "content-component",
  components: {
    EnableEditor: di,
    InlinedScript: tt,
    ContentStyles: mi,
    Blocks: M,
    DynamicDiv: Q
  },
  props: [
    "content",
    "customComponents",
    "data",
    "locale",
    "context",
    "canTrack",
    "apiKey",
    "apiVersion",
    "blocksWrapper",
    "blocksWrapperProps",
    "model",
    "enrich",
    "showContent",
    "contentWrapper",
    "contentWrapperProps",
    "trustedHosts",
    "isSsrAbTest",
    "isNestedRender",
    "linkComponent"
  ],
  data() {
    var e, t;
    return {
      scriptStr: Rr({
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-non-null-asserted-optional-chain
        variationId: (e = this.content) == null ? void 0 : e.testVariationId,
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-non-null-asserted-optional-chain
        contentId: (t = this.content) == null ? void 0 : t.id
      }),
      registeredComponents: [
        ...Te(),
        ...this.customComponents || []
      ].reduce(
        (n, { component: o, ...r }) => ({
          ...n,
          [r.name]: {
            component: bi(o),
            ...ue(r)
          }
        }),
        {}
      ),
      builderContextSignal: {
        content: fi({
          content: this.content,
          data: this.data
        }),
        localState: void 0,
        rootState: hi({
          content: this.content,
          data: this.data,
          locale: this.locale
        }),
        rootSetState: this.contentSetState,
        context: this.context || {},
        canTrack: this.canTrack,
        apiKey: this.apiKey,
        apiVersion: this.apiVersion,
        componentInfos: [
          ...Te(),
          ...this.customComponents || []
        ].reduce(
          (n, { component: o, ...r }) => ({
            ...n,
            [r.name]: ue(r)
          }),
          {}
        ),
        inheritedStyles: {},
        BlocksWrapper: this.blocksWrapper || "div",
        BlocksWrapperProps: this.blocksWrapperProps || {}
      },
      TARGET: A
    };
  },
  provide() {
    const e = this;
    return {
      [Ve.key]: {
        registeredComponents: e.registeredComponents
      }
    };
  },
  methods: {
    contentSetState(e) {
      this.builderContextSignal.rootState = e;
    }
  }
});
function yi(e, t, n, o, r, i) {
  const s = g("InlinedScript"), l = g("ContentStyles"), a = g("Blocks"), u = g("EnableEditor");
  return d(), k(u, S({
    content: e.content,
    data: e.data,
    model: e.model,
    context: e.context,
    apiKey: e.apiKey,
    canTrack: e.canTrack,
    locale: e.locale,
    enrich: e.enrich,
    showContent: e.showContent,
    builderContextSignal: e.builderContextSignal,
    contentWrapper: e.contentWrapper,
    contentWrapperProps: e.contentWrapperProps,
    trustedHosts: e.trustedHosts
  }, {}), {
    default: W(() => {
      var h, c, m, C, y, F, X;
      return [
        e.isSsrAbTest ? (d(), k(s, {
          key: 0,
          id: "builderio-variant-visibility",
          scriptStr: e.scriptStr
        }, null, 8, ["scriptStr"])) : v("", !0),
        e.TARGET !== "reactNative" ? (d(), k(l, {
          key: 1,
          isNestedRender: e.isNestedRender,
          contentId: (h = e.builderContextSignal.content) == null ? void 0 : h.id,
          cssCode: (m = (c = e.builderContextSignal.content) == null ? void 0 : c.data) == null ? void 0 : m.cssCode,
          customFonts: (y = (C = e.builderContextSignal.content) == null ? void 0 : C.data) == null ? void 0 : y.customFonts
        }, null, 8, ["isNestedRender", "contentId", "cssCode", "customFonts"])) : v("", !0),
        $(a, {
          blocks: (X = (F = e.builderContextSignal.content) == null ? void 0 : F.data) == null ? void 0 : X.blocks,
          context: e.builderContextSignal,
          registeredComponents: e.registeredComponents,
          linkComponent: e.linkComponent
        }, null, 8, ["blocks", "context", "registeredComponents", "linkComponent"])
      ];
    }),
    _: 1
  }, 16, ["content", "data", "model", "context", "apiKey", "canTrack", "locale", "enrich", "showContent", "builderContextSignal", "contentWrapper", "contentWrapperProps", "trustedHosts"]);
}
const ki = /* @__PURE__ */ b(gi, [["render", yi]]), vi = f({
  name: "content-variants",
  components: {
    InlinedScript: tt,
    InlinedStyles: ge,
    ContentComponent: ki
  },
  props: [
    "canTrack",
    "content",
    "isNestedRender",
    "model",
    "data",
    "context",
    "apiKey",
    "apiVersion",
    "customComponents",
    "linkComponent",
    "locale",
    "enrich",
    "blocksWrapper",
    "blocksWrapperProps",
    "contentWrapper",
    "contentWrapperProps",
    "trustedHosts"
  ],
  data() {
    return {
      shouldRenderVariants: wr({
        canTrack: G(this.canTrack),
        content: this.content
      }),
      TARGET: A,
      getInitVariantsFnsScriptString: Ir,
      getVariants: _
    };
  },
  mounted() {
  },
  updated() {
  },
  computed: {
    updateCookieAndStylesScriptStr() {
      var e;
      return $r(
        _(this.content).map((t) => ({
          id: t.testVariationId,
          testRatio: t.testRatio
        })),
        ((e = this.content) == null ? void 0 : e.id) || ""
      );
    },
    hideVariantsStyleString() {
      return _(this.content).map((e) => `.variant-${e.testVariationId} { display: none; } `).join("");
    },
    defaultContent() {
      var e;
      return this.shouldRenderVariants ? {
        ...this.content,
        testVariationId: (e = this.content) == null ? void 0 : e.id
      } : Un({
        item: this.content,
        canTrack: G(this.canTrack)
      });
    }
  }
});
function Ci(e, t, n, o, r, i) {
  const s = g("InlinedScript"), l = g("InlinedStyles"), a = g("ContentComponent");
  return d(), p(I, null, [
    !e.isNestedRender && e.TARGET !== "reactNative" ? (d(), k(s, {
      key: 0,
      id: "builderio-init-variants-fns",
      scriptStr: e.getInitVariantsFnsScriptString()
    }, null, 8, ["scriptStr"])) : v("", !0),
    e.shouldRenderVariants ? (d(), p(I, { key: 1 }, [
      $(l, {
        id: "builderio-variants",
        styles: e.hideVariantsStyleString
      }, null, 8, ["styles"]),
      $(s, {
        id: "builderio-variants-visibility",
        scriptStr: e.updateCookieAndStylesScriptStr
      }, null, 8, ["scriptStr"]),
      (d(!0), p(I, null, N(e.getVariants(e.content), (u, h) => (d(), k(a, {
        key: u.testVariationId,
        isNestedRender: e.isNestedRender,
        content: u,
        showContent: !1,
        model: e.model,
        data: e.data,
        context: e.context,
        apiKey: e.apiKey,
        apiVersion: e.apiVersion,
        customComponents: e.customComponents,
        linkComponent: e.linkComponent,
        canTrack: e.canTrack,
        locale: e.locale,
        enrich: e.enrich,
        isSsrAbTest: e.shouldRenderVariants,
        blocksWrapper: e.blocksWrapper,
        blocksWrapperProps: e.blocksWrapperProps,
        contentWrapper: e.contentWrapper,
        contentWrapperProps: e.contentWrapperProps,
        trustedHosts: e.trustedHosts
      }, null, 8, ["isNestedRender", "content", "model", "data", "context", "apiKey", "apiVersion", "customComponents", "linkComponent", "canTrack", "locale", "enrich", "isSsrAbTest", "blocksWrapper", "blocksWrapperProps", "contentWrapper", "contentWrapperProps", "trustedHosts"]))), 128))
    ], 64)) : v("", !0),
    $(a, S({
      isNestedRender: e.isNestedRender,
      content: e.defaultContent,
      showContent: !0,
      model: e.model,
      data: e.data,
      context: e.context,
      apiKey: e.apiKey,
      apiVersion: e.apiVersion,
      customComponents: e.customComponents,
      linkComponent: e.linkComponent,
      canTrack: e.canTrack,
      locale: e.locale,
      enrich: e.enrich,
      isSsrAbTest: e.shouldRenderVariants,
      blocksWrapper: e.blocksWrapper,
      blocksWrapperProps: e.blocksWrapperProps,
      contentWrapper: e.contentWrapper,
      contentWrapperProps: e.contentWrapperProps,
      trustedHosts: e.trustedHosts
    }, {
      key: e.shouldRenderVariants.toString()
    }), null, 16, ["isNestedRender", "content", "model", "data", "context", "apiKey", "apiVersion", "customComponents", "linkComponent", "canTrack", "locale", "enrich", "isSsrAbTest", "blocksWrapper", "blocksWrapperProps", "contentWrapper", "contentWrapperProps", "trustedHosts"])
  ], 64);
}
const Si = /* @__PURE__ */ b(vi, [["render", Ci]]), wi = async ({
  builderContextValue: e,
  symbol: t
}) => {
  if (t != null && t.model && // This is a hack, we should not need to check for this, but it is needed for Svelte.
  (e != null && e.apiKey))
    return ve({
      model: t.model,
      apiKey: e.apiKey,
      apiVersion: e.apiVersion,
      ...(t == null ? void 0 : t.entry) && {
        query: {
          id: t.entry
        }
      }
    }).catch((n) => {
      B.error("Could not fetch symbol content: ", n);
    });
}, Ti = f({
  name: "builder-symbol",
  components: { ContentVariants: Si, DynamicDiv: Q },
  props: [
    "symbol",
    "attributes",
    "dynamic",
    "builderContext",
    "builderBlock",
    "builderComponents",
    "builderLinkComponent"
  ],
  data() {
    var e;
    return { contentToUse: (e = this.symbol) == null ? void 0 : e.content, filterAttrs: x };
  },
  mounted() {
    this.setContent();
  },
  watch: {
    onUpdateHook0: {
      handler() {
        this.setContent();
      },
      immediate: !0
    }
  },
  computed: {
    blocksWrapper() {
      return "div";
    },
    contentWrapper() {
      return "div";
    },
    className() {
      var e, t;
      return [
        this.attributes[ee()],
        "builder-symbol",
        (e = this.symbol) != null && e.inline ? "builder-inline-symbol" : void 0,
        (t = this.symbol) != null && t.dynamic || this.dynamic ? "builder-dynamic-symbol" : void 0
      ].filter(Boolean).join(" ");
    },
    onUpdateHook0() {
      return {
        0: this.symbol
      };
    }
  },
  methods: {
    setContent() {
      this.contentToUse || wi({
        symbol: this.symbol,
        builderContextValue: this.builderContext
      }).then((e) => {
        e && (this.contentToUse = e);
      });
    }
  }
});
function Ii(e, t, n, o, r, i) {
  var l, a, u, h, c;
  const s = g("ContentVariants");
  return d(), p("div", S({ class: e.className }, { ...e.filterAttrs(e.attributes, "v-on:", !1) }, P(e.filterAttrs(e.attributes, "v-on:", !0), !0)), [
    $(s, {
      isNestedRender: !0,
      apiVersion: e.builderContext.apiVersion,
      apiKey: e.builderContext.apiKey,
      context: {
        ...e.builderContext.context,
        symbolId: (l = e.builderBlock) == null ? void 0 : l.id
      },
      customComponents: Object.values(e.builderComponents),
      data: {
        ...(a = e.symbol) == null ? void 0 : a.data,
        ...e.builderContext.localState,
        ...(h = (u = e.contentToUse) == null ? void 0 : u.data) == null ? void 0 : h.state
      },
      canTrack: e.builderContext.canTrack,
      model: (c = e.symbol) == null ? void 0 : c.model,
      content: e.contentToUse,
      linkComponent: e.builderLinkComponent,
      blocksWrapper: e.blocksWrapper,
      contentWrapper: e.contentWrapper
    }, null, 8, ["apiVersion", "apiKey", "context", "customComponents", "data", "canTrack", "model", "content", "linkComponent", "blocksWrapper", "contentWrapper"])
  ], 16);
}
const $i = /* @__PURE__ */ b(Ti, [["render", Ii]]), xe = {};
function Li(e) {
  if (T()) {
    Object.assign(xe, e);
    const t = {
      type: "builder.settingsChange",
      data: xe
    };
    parent.postMessage(t, "*");
  }
}
const Oi = async (e) => {
  var o, r, i;
  const t = e.path || ((o = e.url) == null ? void 0 : o.pathname) || ((r = e.userAttributes) == null ? void 0 : r.urlPath), n = {
    ...e,
    apiKey: e.apiKey,
    model: e.model || "page",
    userAttributes: {
      ...e.userAttributes,
      ...t ? {
        urlPath: t
      } : {}
    },
    options: rt(e.searchParams || ((i = e.url) == null ? void 0 : i.searchParams) || e.options)
  };
  return {
    apiKey: n.apiKey,
    model: n.model,
    content: await ve(n)
  };
};
export {
  Nr as A,
  be as B,
  bn as C,
  Fr as D,
  ve as E,
  kn as F,
  Ei as G,
  Vi as H,
  Rn as I,
  En as S,
  A as T,
  br as V,
  ge as _,
  pn as a,
  xi as b,
  D as c,
  Wi as d,
  fe as e,
  b as f,
  De as g,
  te as h,
  Ai as i,
  ee as j,
  de as k,
  wt as l,
  $i as m,
  fo as n,
  M as o,
  Si as p,
  R as q,
  Lr as r,
  yr as s,
  ei as t,
  Li as u,
  rt as v,
  bi as w,
  Ni as x,
  Fi as y,
  Oi as z
};
